---
title: "Analysis_main"
author: "TR"
date: "`r Sys.Date()`"
output: html_document
---
**This R markdown file contains all relevant analysis steps of the manuscript titled "Ecotype diversification as mechanism for coexistence in extreme environments – the case of northern pike (Esox lucius) in brackish lagoons", including data preparation of elemental transect data, genetics data and growth data (growth data was produced via OtoJ macro of imageJ´s objectJ plugin). Not included are the assignment of element transect data to individual years, raw data preparation of SIMS and LA-ICPMS analysis and plotting of individual-level transects for evaluation of data quality, outliers or vaterite inclusions (indicated by Sr - Mg interaction)**

**All chunks except for the packages chunk have been set to eval = FALSE, chunks can be run individually by setting eval = TRUE**

**Note that this file has not been optimized for any kind of output, if you wish to produce an html or word document, you may need to adapt figure and table outputs manually**

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r packages, include=FALSE}
library(tidyverse)
library(ggplot2)
library(cowplot)
library(ggridges)
library(lme4)
library(nlme)
library(lmerTest)
library(emmeans)
library(MuMIn)
library(visreg)
library(car)
library(sjPlot)
library(knitr)
library(car)
library(ggpubr)
library(mclust)
library(dtwclust)
library(vegan)
library(mvabund)
library(RVAideMemoire)
library(MASS)
library(flextable)
library(gridGraphics)
library(lubridate)
library(factoextra)
library(data.table)
library(gridExtra)
library(loo)
library(bayesplot)
library("rstan")
options(mc.cores = parallel::detectCores())
rstan_options(auto_write = TRUE)
```

# Temperature and salinity in the study area
## Environmental data
```{r ipdw loop data, include=FALSE, eval=FALSE}
#Scrit to read and process environmental data from mutliple sources
rm(list = ls())

#LUNGdata: Data measured by Landesamt für Umwelt, naturschutz & Geologie Mecklenburg-Vorpommern (LUNG MV)
#https://www.lung.mv-regierung.de/ and https://www.umweltkarten.mv-regierung.de/
#Data time span: 2003 - 2022 monthly measurments (some stations biweekly, in which case values were averaged for month)
LUNGdata <- read.delim(file = "Data/environment/LUNGdata_clean.txt", sep = "\t", header = T, stringsAsFactors = F)

#Samplingdata: Opportunistic measurements of temperature & salinity taken in the course of sampling events of 
#project Boddenhecht in the time from 2020 to 2022 (irregular intervals)
Samplingdata <- read.delim(file = "Data/environment/Samp_Inst_months.txt", sep = "\t", header = T, stringsAsFactors = F)

#Loggerdata: Measurments of temperature and salinty taken by HOBO logger attached acoustic receivers deployed in the
#study area, for more information on receiver network, see Dhellemmes et al. (2023),
#https://doi.org/10.1016/j.fishres.2022.106560
Loggerdata <- read.delim(file = "Data/environment/Logger_month_means.txt", sep = "\t", header = T, stringsAsFactors = F)

#Bring all into same format, LUNG data
LUNGdata <- LUNGdata%>%
  ungroup()%>%
  transmute(name = name,
            id = id,
            waterbody = case_when(waterbody=="Sehrower Bach"~"Sehrowbach",
                                  waterbody=="Westziese"~"Ziese",
                                  waterbody=="Ostziese"~"Ziese",
                                  waterbody=="Duvenbaek"~"Duwenbeek",
                                  TRUE~waterbody),
            month = month,
            year = year,
            Lat = Lat,
            Long = Long,
            type = type,
            turbidity=turbidity,
            SAL = case_when(waterbody=="Sehrowbach"~mean(Samplingdata[Samplingdata$waterbody=="Sehrowbach",]$SAL),
                            waterbody=="Barthe"~mean(Samplingdata[Samplingdata$waterbody=="Barthe",]$SAL),
                            waterbody=="Ziese"~mean(Samplingdata[Samplingdata$waterbody=="Ziese",]$SAL),
                            waterbody=="Recknitz"~mean(Samplingdata[Samplingdata$waterbody=="Recknitz",]$SAL),
                            waterbody=="Duwenbeek"~mean(Samplingdata[Samplingdata$waterbody=="Duwenbeek",]$SAL),
                            waterbody=="Ryck"~mean(Samplingdata[Samplingdata$waterbody=="Ryck",]$SAL),
                            TRUE~SAL),
            WT = WT)

#BH sampling
Samplingdata <- Samplingdata%>%
  transmute(name = name,
            id = id,
            waterbody = waterbody,
            month = month,
            year = year,
            Lat = Lat,
            Long = Long,
            type = type,
            turbidity=NA,
            SAL = SAL,
            WT = WT)

#Logger
Loggerdata <- Loggerdata%>%
  transmute(name = name,
            id = id,
            waterbody = waterbody,
            month = month,
            year = year,
            Lat = Lat,
            Long = Long,
            type = type,
            turbidity=NA,
            SAL = SAL,
            WT = WT)

#Filter LUNGdata into lagoons and tributaries
Tributaries <- LUNGdata%>%
  filter(waterbody=="Sehrowbach"|waterbody=="Barthe"|waterbody=="Ziese"|waterbody=="Peene")
Lagoons <- LUNGdata%>%
  filter(type=="Coast")

#filter sampling data into lagoons and tributaries
Tributaries2 <- Samplingdata%>%
  filter(waterbody%in%c("Sehrowbach","Barthe","Ziese","Peene","Badendycksgraben","NHG"))
Lagoons2 <- Samplingdata%>%
  filter(!waterbody%in%c("Sehrowbach","Barthe","Ziese","Peene","Badendycksgraben","NHG","Beek","Duwenbeek","Peene","Ryck"))

#filter logger data into lagoons and tributaries
Tributaries3 <- Loggerdata%>%
  filter(waterbody%in%c("Sehrowbach","Barthe","Ziese","Peene","Badendycksgraben","NHG"))
Lagoons3 <- Loggerdata%>%
  filter(!waterbody%in%c("Sehrowbach","Barthe","Ziese","Peene","Badendycksgraben","NHG","Beek","Duwenbeek","Peene","Ryck"))

#Join
Tributaries <- rbind(Tributaries, Tributaries2, Tributaries3)
Lagoons <- rbind(Lagoons, Lagoons2, Lagoons3)

fwrite(Tributaries, "Data/environment/Environmental-params-tributary.csv")
fwrite(Lagoons, "Data/environment/Environmental-params-lagoon.csv")
```

```{r ipdw loop, include=FALSE, eval=FALSE}
#Script to interpolate environmental parameters in the study area using inverse path distance weighted (ipdw) interpolation
rm(list = ls())

Tributaries <- fread("Data/environment/Environmental-params-tributary.csv")
Lagoons <- fread("Data/environment/Environmental-params-lagoon.csv")

#We performed the interpolation only with lagoon data, as including the tributaries resulted in unrealisticly high influence
#of small tributaries (e.g., NHG) on lagoon salinities
Saldata <- Lagoons%>%
  filter(year >= 2005)%>%
  group_by(name, month, year)%>%
  reframe(name=name,
          waterbody=waterbody,
          year=year,
          month=month,
          identifier=paste0(as.character(year), "-", as.character(month)),
          Lat=as.numeric(Lat),
          Long=as.numeric(Long),
          type=type,
          turbidity=round(mean(turbidity),2),
          SAL=round(mean(SAL),2),
          WT=round(mean(WT),2))%>%
  distinct()

#correct typo in sampling point
Saldata[Saldata$identifier=="2022"&Saldata$waterbody=="SB",]$Long <- 13.15835

#load in receiver for later checking accessibility of rasterstack
# Rec <- read.csv2("Data/Rec-Arrays_ALL.csv", header = T, 
#                  stringsAsFactors = F, sep = ",")
# Rec$Longitude <- as.numeric(Rec$Longitude)
# Rec$Latitude <- as.numeric(Rec$Latitude)
# rec_xy <- Rec[,1:2]
# rec_sp <- SpatialPointsDataFrame(coords = rec_xy, data = Rec, 
#                                         proj4string = CRS("+init=epsg:4326"))
# rec_sp <- spTransform(rec_sp, CRS("+proj=utm +zone=33 +ellps=GRS80 
#                                   +units=m +no_defs"))

library(rgdal)
library(raster)
library(ipdw)
library(spatstat)
library(sf)

#load in land layer geopackage
Land <- readOGR("Data/environment/Studyarea_buffered2.gpkg")
  
#transform to UTM (meters)
Land <- spTransform(Land, CRS("+proj=utm +zone=33 +ellps=GRS80 +units=m +no_defs"))

#Prepare ipdw loop, one identifier per month for each year 2005 - 2022
monthID <- unique(Saldata$identifier)[1]

#empty stacks to fill with interpolated rasters
raster <- stack()
mras <- stack()

#Validation object
validation <- NULL

for (i in 1:length(monthID)) {
    #select data (month & year through identifier)
    df <- Saldata[which(Saldata$identifier == monthID[i]),]
    
    #generate coordinates, make sure format is Long-Lat
    xy <- df[,c(7,6)]
    
    #convert to spatial points
    pnts <- SpatialPointsDataFrame(coords = xy, data = df,
                                   proj4string = CRS("+init=epsg:4326"))
    pnts <- spTransform(pnts, CRS("+proj=utm +zone=33 +ellps=GRS80 +units=m +no_defs"))
    
    #generate costraster, resolution in m
    costras <- costrasterGen(pnts, Land, extent = "pnts", 
                             resolution = 250,
                             projstr = projection(pnts))
    
    #mean neighbor distance
    W <- owin(range(coordinates(pnts)[,1]), range(coordinates(pnts)[,2]))
    kat.pp <- ppp(coordinates(pnts)[,1], coordinates(pnts)[,2], window = W)
    mean.neighdist <- mean(nndist(kat.pp))
    
    #build grid
    gridsize <- mean.neighdist*2
    grainscale.fac <- gridsize/res(costras)[1]
    gridras <- aggregate(costras, fact = grainscale.fac)
    gridpol <- rasterToPolygons(gridras)
    gridpol$value <- row.names(gridpol)
    
    #spatial join
    fulldataset.over <- over(pnts, gridpol)
    fulldataset.over <- cbind(data.frame(fulldataset.over), 
                          setNames(data.frame(pnts), c("name",
                                                       "month",
                                                       "year",
                                                       "waterbody",
                                                       "identifier",
                                                       "Lat", "Long", "type",
                                                       "turbidity", "SAL", "WT",
                                                       "x.utm","y.utm")))
    #grid selection
   set.seed(2)
    gridlev <- unique(fulldataset.over$value)
    #split in training & validation data sets
    for (i in seq_along(gridlev)) {
      activesub <- subset(fulldataset.over, fulldataset.over$value==gridlev[i])
      selectnum <- gdata::resample(seq_len(nrow(activesub)), 1)
      if(i == 1){
        training <- activesub[selectnum,]
        }
      else{
        training <- rbind(training, activesub[selectnum,])
      }
    }
    
    training <- st_as_sf(training, coords = c("x.utm", "y.utm"))
    
    paramlist <- c("SAL")
    
    #perform ipdw, set name to month & year
    ipdw <- ipdw(training, costras, range = mean.neighdist*10, 
                       paramlist, overlapped = TRUE)
    ipdw <- setNames(ipdw[[1]], paste(unique(df$year), "Month",unique(df$month)))
    
    #error estimation, generate data frame containing measured & interpolated
    measured.spdf <- data.frame(validate@data[11:14])
    coordinates(measured.spdf) <- coordinates(validate)
    
    #claculate simple measurement error metrics, name after year & month
    valid.ipdw <- errorGen(ipdw, measured.spdf, measured.spdf@data$SAL)
    valid.ipdw$id <- paste(unique(df$year),"Month", unique(df$month))
    
    #save measurement errors in list
    validation <- c(validation, valid.ipdw)
    #save interpolations into rasterstack
    raster <- stack(raster, ipdw)
}

plot(Land)
points(pnts)
plot(costras)
points(pnts)
plot(raster$X2019.Month.1.1.1)

plot(sras2$Spring_2021)
points(pnts)

 #example for simple error comparison between prediction & measured
valid <- lm(validation[[2]]$SAL ~ validation[[2]]$predicted)

summary(valid)

plot(validation[[2]]$SAL, validation[[2]]$predicted)

errorGen(ipdw, measured.spdf, measured.spdf@data$SAL,
  plot = TRUE, title = "IPDW")

#This extracts salinity values from around the receivers and forms a mean

raster::extract(yras, rec_sp, buffer = 500, small = T, fun = mean, na.rm = T)

salras <- tras

#save to RData
save(sras, file = "Data/Spatial/Output/Translocation_salinity_2021_lowrange.RData")
save(tras, file = "Data/Spatial/Output/Temp-IPDW-mean-2007-2022_monthly.RData")
save(raster, file = "Data/Spatial/Output/Turb-IPDW-mean-2007-2022_monthly.RData")

#monthly salinity
load("Data/Spatial/Output/Salinity-IPDW-all_07-22.RData")
#monthly temperature
load("Data/Spatial/Output/Temp-IPDW-mean-2007-2022_monthly.RData")
#monthly turbidity
load("Data/Spatial/Output/Turb-IPDW-mean-2007-2022_monthly.RData")

mean_sal <- calc(raster, fun = mean)
ssd <- calc(raster, fun = sd)
srange <- calc(sras, fun = range)
smax <- calc(sras, fun = max)

srange_upper <- srange$layer.2

plot(tras$X2022.Month.10)
plot(ssd)
plot(srange_absolute)
plot(smax)

#yearly temperature
tmean <- calc(tras, fun = mean)
trange <- calc(tras, fun = range)
sdRast <- calc(tras, fun = sd)

trange_absolute <- trange$layer.2-trange$layer.1

plot(raster)

#Write salinity rasters
writeRaster(mean_sal, "Data/Spatial/Output/Salinity_raster_book_revised", format="GTiff", overwrite=T)
writeRaster(ssd, "Data/Spatial/Output/Salinity_SD_book_revised", format="GTiff", overwrite=T)
writeRaster(srange_upper,"Data/Spatial/Output/Sal-ipdw_range_07-22", format="GTiff", overwrite=T)
writeRaster(smax,"Data/Spatial/Output/Sal-ipdw_max_07-22", format="GTiff", overwrite=T)

#Write Felirasters
writeRaster(sras$Spring_2021, "Data/Spatial/Output/Translocation_salinity_2021_lowrange", format="raster",
            overwrite=T)
writeRaster(sras2$Spring_2021, "Data/Spatial/Output/Translocation_salinity_2021_highrange", format="raster",
            overwrite=T)



#Write temperature rasters
writeRaster(srange_absolute, "Data/Spatial/Output/Sal-ipdw_range_07-22", format="GTiff")
writeRaster(ssd, "Data/Spatial/Output/Sal-ipdw_sd_07-22", format="GTiff", overwrite=T)
writeRaster(smean, "Data/Spatial/Output/Sal-ipdw_mean_07-22", format="GTiff", overwrite=T)


#Write measuring station shapefile
rgdal::writeOGR(pnts, dsn = "Data/Spatial/Output", layer = "coords", driver = "ESRI Shapefile")
plot(pnts)
```
# Life history phenotypes
## Data prep for niche analysis
### Notes on preparation of Sr and d18O data
Five fish excluded: 
"BH-01701" -> Misplaced transect,
"BH-01741" -> Vaterite,
"BH-01908" -> Vaterite,
"BH-90962" -> Lost during SIMS preprocessing,
"BH-93190" -> no SIMS transect (lightning strike occurred during analysis)
```{r marry d18O data to Sr:Ca data, include=FALSE, eval=FALSE}
#Script to align Sr:Ca values (5.5 um step size) with d18O values (35 um step size) in a common data frame
rm(list = ls())

#Read in LA-ICPMS and SIMS data
LA <- fread("Data/pike/elements/Pike-LA-ICPMS-years_whole-sample_with-TL.txt")
d18O <- fread("Data/pike/elements/Pike-d18O-years_whole-sample_with-TL.txt")

#Clean up data & assign capture locations to larger study areas
#Clean up Trace element data
LA_clean <- LA%>%
  filter(LA_dist>=50)%>%
  filter(comment=="OK")%>%
  transmute(fish_ID = fishID,
            age=age,
            capt = area,
            area = case_when(area %in% "Peene" ~ "Freshwater",
                             area %in% "Barthe" ~ "Freshwater",
                             area %in% "NHG" ~ "Freshwater",
                             area %in% "Sehrowbach" ~ "Freshwater",
                             area %in% "Bandycksgraben" ~ "Freshwater",
                             area %in% "Ziese" ~ "Freshwater",
                             area %in% "KB" ~ "WRB",
                             area %in% "SB" ~ "WRB",
                             area %in% "WB" ~ "NRB",
                             area %in% "BEG" ~ "NRB", 
                             area %in% "KJB" ~ "NRB",
                             area %in% "GJB" ~ "NRB",
                             area %in% "GB" ~ "GB",
                             area %in% "KD" ~ "control",
                             TRUE ~ "NA"),
            Sr = Sr,
            Na = Na,
            Mg = Mg,
            Ba = Ba,
            Mn = Mn,
            LA_dist = LA_dist,
            lifeyear=lifeyear)

#Clean up d18O data
d18O_clean <- d18O%>%
  filter(comment=="OK")%>%
  transmute(fish_ID = fishID,
            age=age,
            capt = area,
            area = case_when(area %in% "Peene" ~ "Freshwater",
                             area %in% "Barthe" ~ "Freshwater",
                             area %in% "NHG" ~ "Freshwater",
                             area %in% "Sehrowbach" ~ "Freshwater",
                             area %in% "Badendycksgraben" ~ "Freshwater",
                             area %in% "Ziese" ~ "Freshwater",
                             area %in% "KB" ~ "WRB",
                             area %in% "SB" ~ "WRB",
                             area %in% "WB" ~ "NRB",
                             area %in% "BEG" ~ "NRB", 
                             area %in% "KJB" ~ "NRB",
                             area %in% "GJB" ~ "NRB",
                             area %in% "GB" ~ "GB",
                             area %in% "KD" ~ "Control",
                             TRUE ~ "NA"),
            d18O = d18O,
            sims_dist = sims_dist,
            lifeyear=lifeyear)

LA_clean <- LA_clean %>% drop_na()

#Smooth trace element data to remove noise
LA_smooth <- LA_clean%>%group_by(fish_ID)%>%
  transmute(fish_ID=fish_ID,
            age=age,
            capt=capt,
            area=area,
            Sr=as.numeric(zoo::rollmean(Sr, k=9, fill=NA, align = "left")),
            Na=as.numeric(zoo::rollmean(Na, k=9, fill=NA, align = "left")),
            Mg=as.numeric(zoo::rollmean(Mg, k=9, fill=NA, align = "left")),
            Ba=as.numeric(zoo::rollmean(Ba, k=9, fill=NA, align = "left")),
            Mn=as.numeric(zoo::rollmean(Mn, k=9, fill=NA, align = "left")),
            LA_dist=as.numeric(LA_dist),
            lifeyear=lifeyear)

#Order data sets for joining
d18O_clean <- d18O_clean[order(d18O_clean$fish_ID),]
LA_smooth <- LA_smooth[order(LA_smooth$fish_ID),]
LA_smooth <- data.frame(LA_smooth)

#Convert to data frame, so approx (), as approx() behaves weird on data tables
d18O_clean <- as.data.frame(d18O_clean)
LA_clean <- as.data.frame(LA_clean)

#reinterpolate Sr (higher spatial resolution ~5.5 um) to d18O distances (lower spatial resolution, ~35 um) before joining
IDOto <- unique(d18O_clean$fish_ID)
dfE <- data.frame()
for (i in 1:length(IDOto)) {
  Sr <- data.frame(approx(LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"LA_dist"],
              LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"Sr"],
              n = length(d18O_clean[which(d18O_clean$fish_ID==IDOto[i]),"sims_dist"]),
              rule = 1, method = "linear", ties = "ordered"))
  Na <- data.frame(approx(LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"LA_dist"],
              LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"Na"],
              n = length(d18O_clean[which(d18O_clean$fish_ID==IDOto[i]),"sims_dist"]),
              rule = 1, method = "linear", ties = "ordered"))
  Mg <- data.frame(approx(LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"LA_dist"],
              LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"Mg"],
              n = length(d18O_clean[which(d18O_clean$fish_ID==IDOto[i]),"sims_dist"]),
              rule = 1, method = "linear", ties = "ordered"))
  Ba <- data.frame(approx(LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"LA_dist"],
              LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"Ba"],
              n = length(d18O_clean[which(d18O_clean$fish_ID==IDOto[i]),"sims_dist"]),
              rule = 1, method = "linear", ties = "ordered"))
  Mn <- data.frame(approx(LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"LA_dist"],
              LA_smooth[which(LA_smooth$fish_ID==IDOto[i]),"Mn"],
              n = length(d18O_clean[which(d18O_clean$fish_ID==IDOto[i]),"sims_dist"]),
              rule = 1, method = "linear", ties = "ordered"))
  df <- cbind(Sr=Sr[,2], Na=Na[,2], Mg=Mg[,2], Ba=Ba[,2], Mn=Mn[,2])
  dfE <- rbind(dfE,df)
}

oto.int <- cbind(d18O_clean, dfE)

fwrite(oto.int, file = "Data/pike/elements/Oto_LA-to-d18O_interpolation.csv")
```

```{r covariance analysis d18O vs Sr and data correction, include=FALSE, eval=FALSE}
#Script to correct d18O values for salinity effects by transforming to residuals from linear regression d18O~Sr:Ca
rm(list = ls())

#read in combined element transect data
oto.int <- fread("Data/pike/elements/Oto_LA-to-d18O_interpolation.csv")

#clean up to remove fish from control lake (but make subset for later join)
oto.int2 <- oto.int%>%
  filter(capt!="KD")%>%
  drop_na()

subset_KD <- subset(oto.int, capt=="KD")

#plot correlation between d18O and Sr
Correlation.Sr <- ggplot(oto.int2,aes(x=Sr*1000, y=d18O))+
  geom_point(size=.5)+
  scale_x_continuous(breaks = seq(0,8,1))+
  scale_y_continuous(breaks = seq(-15,0,1))+
  labs(x="Sr/Ca (mg/g)", 
       y=expression(bold(delta^18*"O"~("VPDB \u2030"))),
       title=expression(bold("Correlation of intraotolith Sr and"~delta^18*"O")))+
  stat_smooth(method = "lm", fullrange = T, size = .8)+
  theme_classic()+
    theme(panel.grid.major = element_line(),
          axis.line = element_line(size = .5, colour = "black", linetype = 1),
          axis.ticks = element_line(size = .5, colour = "black"),
          axis.text = element_text(size = 10, colour = "black"),
          axis.title.x = element_text(size = 10, face = "bold"),
          axis.title.y = element_text(size = 10, face = "bold"),
          legend.text = element_text(size = 5),
          legend.title = element_text(size = 10, face = "bold"),
          plot.title = element_text(size = 15, face = "bold", hjust = .5))

ggsave("Figures/supplement/Correlation_d18O~Sr.png", Correlation.Sr, width = 15, height = 9, units = "cm", dpi = 600, bg="white")

#Test different models
#linear model
lm1 <- lm(d18O~Sr, data = oto.int2)
#exponential decay model
lm2 <- lm(d18O~exp(-Sr), data = oto.int2)

#Test models
anova(lm1, lm2) #exponential decay is not significantly better than a linear model

#summarize final model
summary(lm1)

#Extract residuals
oto.int2$res <- lm1$residuals

#fill in scaled values for d18O in pike from control lake
subset_KD$res <- scale(subset_KD$d18O)
subset_KD <- subset_KD%>%drop_na()

oto.int3 <- rbind(oto.int2, subset_KD)

#check for pattern retention
par(mfrow = c(1,2))
plot(oto.int3[oto.int3$fish_ID=="BH-01906",]$sims_dist, oto.int3[oto.int3$fish_ID=="BH-01906",]$d18O, type="l",
     xlab = "Distance from core (um)", ylab = "d18O (permil VPDB)")
plot(oto.int3[oto.int3$fish_ID=="BH-01906",]$sims_dist, oto.int3[oto.int3$fish_ID=="BH-01906",]$res, type="l",
     xlab = "Distance from core (um)", ylab = "d18O residual")

fwrite(oto.int3, "Data/pike/elements/Oto_LA-to-d18O_interpolation_Sr-corrected.csv")
```

```{r average corrected d18O over years, include=FALSE, eval=FALSE}
#Script to average over yearly temperature signal in d18O to arrive at mean lifelong thermal niche
rm(list = ls())

#read in data
oto.int <- fread("Data/pike/elements/Oto_LA-to-d18O_interpolation_Sr-corrected.csv")

#Make lifeyear a factor
oto.int$lifeyear <- as.factor(oto.int$lifeyear)

#calculate average number of d18O points per year
yearmean <- oto.int%>%group_by(fish_ID,lifeyear)%>%summarise(n=n())
mean(yearmean$n) #mean of 7.8 --> smoothing data with 7 points rolling mean

oto.int2 <- oto.int%>%
  group_by(fish_ID)%>%
  transmute(fish_ID=fish_ID,
            age=age,
            capt=capt,
            area=area,
            d18O=zoo::rollmean(d18O, k=7, fill = NA, align = "left"),
            res=zoo::rollmean(res, k=7, fill = NA, align = "left"),
            Sr=Sr,
            Na=Na,
            Mg=Mg,
            Ba=Ba,
            Mn=Mn,
            year=lifeyear,
            dist=sims_dist)

#check whether smoothing averaged over seasonal d18O signal
par(mfrow=c(1,2))
plot(oto.int[oto.int$fish_ID=="BH-01855",]$sims_dist, oto.int[oto.int$fish_ID=="BH-01855",]$d18O, type="l")
plot(oto.int2[oto.int2$fish_ID=="BH-01855",]$dist, oto.int2[oto.int2$fish_ID=="BH-01855",]$d18O, type="l")

#Save data
fwrite(oto.int2, "Data/pike/elements/Oto_LA-to-d18O_interpolation_Sr-corrected_smooth.csv")
```
## Lifelong niche (individual and population level)
```{r calculate profile plots for individuals, include=FALSE, eval=FALSE}
#Script to calculate individual-level life history plots for individual visual assessment of thermosaline history
rm(list = ls())

#read in raw data
LA <- fread("Data/pike/elements/Pike-LA-ICPMS-years_whole-sample_with-TL.txt")
d18O <- fread("Data/pike/elements/Pike-d18O-years_whole-sample_with-TL.txt")

#Convert to data frame
LA <- as.data.frame(LA)
d18O <- as.data.frame(d18O)

#Format and simplify capture locations
LA <- LA%>%
  transmute(ID = fishID,
            capt = area,
            area = case_when(area %in% "Peene" ~ "Freshwater",
                             area %in% "Barthe" ~ "Freshwater",
                             area %in% "NHG" ~ "Freshwater",
                             area %in% "Sehrowbach" ~ "Freshwater",
                             area %in% "Badendycksgraben" ~ "Freshwater",
                             area %in% "Ziese" ~ "Freshwater",
                             area %in% "KB" ~ "Marine",
                             area %in% "SB" ~ "Marine",
                             area %in% "WB" ~ "Marine",
                             area %in% "BEG" ~ "Marine", 
                             area %in% "KJB" ~ "Marine",
                             area %in% "GJB" ~ "Marine",
                             area %in% "GB" ~ "Marine",
                             area %in% "KD" ~ "control",
                             TRUE ~ "NA"),
            Sr = Sr,
            Na = Na,
            Mg = Mg,
            Ba = Ba,
            Mn = Mn,
            LA_dist = LA_dist)

d18O <- d18O%>%
  transmute(ID = fishID,
            capt = area,
            area = case_when(area %in% "Peene" ~ "Freshwater",
                             area %in% "Barthe" ~ "Freshwater",
                             area %in% "NHG" ~ "Freshwater",
                             area %in% "Sehrowbach" ~ "Freshwater",
                             area %in% "Bandycksgraben" ~ "Freshwater",
                             area %in% "Ziese" ~ "Freshwater",
                             area %in% "KB" ~ "Marine",
                             area %in% "SB" ~ "Marine",
                             area %in% "WB" ~ "Marine",
                             area %in% "BEG" ~ "Marine", 
                             area %in% "KJB" ~ "Marine",
                             area %in% "GJB" ~ "Marine",
                             area %in% "GB" ~ "Marine",
                             area %in% "KD" ~ "Freshwater",
                             TRUE ~ "NA"),
            d18O = d18O,
            sims_dist = sims_dist)


LA <- LA %>% drop_na()
d18O <- d18O%>%drop_na()

# Loop to plot individuals
mypallete <- viridis::viridis(4, direction = 1)

IDOto <- unique(d18O$ID)

for (i in 1:length(IDOto)) {
  png(paste0("Figures/supplement/individual plots/", IDOto[i], ".png"), width = 1200)
  par(mfrow=c(1,2))
  plot(LA[which(LA$ID == IDOto[i]), "LA_dist",], 
       LA[which(LA$ID == IDOto[i]), "Sr",], main = "", 
       xlab =  "", ylab = "", 
       xlim = c(0, max(LA[which(LA$ID == IDOto[i]), "LA_dist",], na.rm = T)),
       ylim = c(0, max(LA$Sr, na.rm = T)),
       col.axis = "black", type = "l", col = "orange", lwd = 4, 
       fg = "black", axes = FALSE, pch = 19)
  axis(1, pos = mean(LA[LA$area == "Freshwater",]$Sr, na.rm = T), 
       at = c(0, max(LA[which(LA$ID == IDOto[i]), "LA_dist",], na.rm = T)-20,
              max(LA$LA_dist)),
       labels = c("", "Freshwater", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[3], lty = 1, lwd = 3, mgp = c(0,0.5,0), 
       cex.axis=2, font = 2)
  axis(1, pos = mean(LA[LA$area == "Freshwater",]$Sr, na.rm = T)+
         sd(LA[LA$area == "Freshwater",]$Sr, na.rm = T), 
       at = c(0, 1200, max(LA$LA_dist)),
       labels = c("", "", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[3], lty = 3, lwd = 3, mgp = c(0,0,0))
  axis(1, pos = mean(LA[LA$area == "Freshwater",]$Sr, na.rm = T)-
         sd(LA[LA$area == "Freshwater",]$Sr, na.rm = T), 
       at = c(0, 1200, max(LA$LA_dist)),
       labels = c("", "", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[3], lty = 3, lwd = 3, mgp = c(0,0,0))
  axis(1, pos = mean(LA[LA$area == "Marine",]$Sr, na.rm = T), 
       at = c(0, max(LA[which(LA$ID == IDOto[i]), "LA_dist",], na.rm = T)-20,
                   max(LA$LA_dist)),
       labels = c("", "Brackish", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[1], lty = 1, lwd = 3, mgp = c(0,0.5,0), 
       cex.axis=2, font = 2)
  axis(1, pos = mean(LA[LA$area == "Marine",]$Sr, na.rm = T)-
         sd(LA[LA$area == "Marine",]$Sr, na.rm = T), 
       at = c(0, 1200, max(LA$LA_dist)),
       labels = c("", "", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[1], lty = 3, lwd = 3, mgp = c(0,0,0))
  axis(1, pos = mean(LA[LA$area == "Marine",]$Sr, na.rm = T)+
         sd(LA[LA$area == "Marine",]$Sr, na.rm = T), 
       at = c(0, 1200, max(LA$LA_dist)),
       labels = c("", "", ""), lwd.ticks = 0, tck = 0, 
       col = mypallete[1], lty = 3, lwd = 3, mgp = c(0,0,0))
  axis(1, pos = 0, at = seq(0, max(LA$LA_dist), 500),
       labels = seq(0, max(LA$LA_dist), 500), lwd.ticks = 5, tck = 0, 
       col = "black", lty = 1, lwd = 2, mgp = c(0, 0.5, 0),
       cex.axis=2, font = 2)
  axis(2, pos = 0, tcl = 0.2,
       at = seq(round(min(LA$Sr),3), 
                round(max(LA$Sr),3), 0.001), 
       labels = seq(round(min(LA$Sr)*1000,0), 
                    round(max(LA$Sr)*1000,0), 1),
       las = 2, mgp = c(0, 0.5, 0), cex.axis = 2, font = 2, 
       lwd = 2, col = "black", col.axis = "black")
  title(ylab = list("Sr/Ca ratio (mg/g)", cex = 2, font=2),
        xlab = list("distance from core [um]", cex = 2, font=2), 
        col.lab = "black", line = 1)
  text(x = 650, y = 0.008, labels = paste(IDOto[i], "-", 
                                         LA[which(LA$ID == IDOto[i]), "capt",]), 
       cex=3, font=2)

  plot(d18O[which(d18O$ID == IDOto[i]), "sims_dist",],
       d18O[which(d18O$ID == IDOto[i]), "d18O",], main = "",
       xlab =  "", ylab = "",
       xlim = c(0, max(d18O[which(d18O$ID == IDOto[i]), "sims_dist",], na.rm = T)),
       ylim = c(min(d18O$d18O, na.rm = T), 0),
       col.axis = "black", cex.axis = 1, type = "b", col = "blue", lwd = 3,
       fg = "black", cex = 1, axes = FALSE, pch = 19)
  axis(1, pos = mean(d18O[which(d18O$ID == IDOto[i]), "d18O",], na.rm = T),
       at = c(0, 200, max(d18O[which(d18O$ID == IDOto[i]), "sims_dist",], na.rm = T)),
       labels = c("", "indiv.mean", ""), lwd.ticks = 0, tck = 0,
       col = "black", lty = 3, lwd = 1, mgp = c(0,0,0))
  axis(1, pos = mean(d18O$d18O, na.rm = T),
       at = c(0, 200, max(d18O$sims_dist)),
       labels = c("", "global mean", ""), lwd.ticks = 0, tck = 0,
       col = "black", lty = 1, lwd = 1, mgp = c(0,0,0))
  axis(1, pos = round(min(d18O$d18O), 1), at = seq(0, max(d18O$sims_dist), 500),
       labels = seq(0, max(d18O$sims_dist), 500), lwd.ticks = 5, tck = 0,
       col = "black", lty = 1, lwd = 2, mgp = c(0, 0.5, 0))
  axis(2, pos = 0, tcl = 0.2,
       at = seq(round(min(d18O$d18O, na.rm = T),1),
                round(max(d18O$d18O, na.rm = T),1), 1),
       labels = seq(round(min(d18O$d18O, na.rm = T),1),
                    round(max(d18O$d18O, na.rm = T),1), 1),
       las = 2, mgp = c(0, 0.5, 0), cex.axis = 1,
       lwd = 2, col = "black", col.axis = "black")
  title(ylab = "d18O", cex.lab = 1,
        col.lab = "black", line = 5)
  title(xlab = "Distance [um]", cex.lab = 1,
        col.lab = "black", line = 1)
  title(main = "d18O")
  text(x = 300, y = 0, labels = d18O[which(d18O$ID == IDOto[i]), "capt",])
  dev.off()  
}
```

## DTW-clustering of otolith timeseries
```{r subset data, include=FALSE, eval=FALSE}
#Script to subset data sets into lagoon and freshwater sample
rm(list = ls())
#read in data
oto.int <- fread("Data/pike/elements/Oto_LA-to-d18O_interpolation_Sr-corrected_smooth.csv")

#remove NAs
oto.int <- na.omit(oto.int)

#subset according to cluster
Fgroup <- subset(oto.int, area=="Freshwater")
Bgroup <- oto.int%>%filter(area=="WRB"|area=="NRB"|area=="GB")

#Test length, needs to equal N=120
length(unique(Bgroup$fish_ID))+length(unique(Fgroup$fish_ID))

#data package
fwrite(Fgroup, "Data/pike/clustering/Tributary-sample-transects.csv")
fwrite(Bgroup, "Data/pike/clustering/Lagoon-sample-transects.csv")
```

```{r dtw data prep and clustering, include=FALSE, eval=FALSE}
#Script to reinterpolate transects to standard length and perform DTW clustering
rm(list = ls())

#read in data
Fgroup <- as.data.frame(fread("Data/pike/clustering/Tributary-sample-transects.csv"))
Bgroup <- as.data.frame(fread("Data/pike/clustering/Lagoon-sample-transects.csv"))

#load funtions
source("funcs.R")

#Look for sample size
length(unique(Fgroup$fish_ID))
length(unique(Bgroup$fish_ID))

#Mean transect length
median(table(Fgroup$fish_ID)) #Median length 35.5 data points
median(table(Bgroup$fish_ID)) #Median length 31.5 data points

#Make lifeyear a factor
Bgroup$lifeyear <- as.factor(Bgroup$year)
Fgroup$lifeyear <- as.factor(Fgroup$year)

#reinterpolate to standard length (based on medians)
dataB1 <- interpol(series = Bgroup, length = 35, element = "Sr")
dataB2 <- interpol(series = Bgroup, length = 35, element = "res")

dataF1 <- interpol(series = Fgroup, length = 35, element = "Sr")
dataF2 <- interpol(series = Fgroup, length = 35, element = "res")

dataB <- cbind(dataB1, dataB2[,2])
dataF <- cbind(dataF1, dataF2[,2])

dataB[,2] <- scale(dataB[,2])
dataF[,2] <- scale(dataF[,2])

#Data Lists for hierarchical clustering
IDOto <- unique(dataB$ID)
Bseries <- list()

for (i in 1:length(IDOto)) {
  m <- as.matrix(dataB[which(dataB$ID==IDOto[i]),c(2,3)])
  Bseries[[length(Bseries)+1]] <- m
}

names(Bseries) <- c(IDOto)

IDOto <- unique(dataF$ID)
Fseries <- list()

for (i in 1:length(IDOto)) {
  m <- as.matrix(dataF[which(dataF$ID==IDOto[i]),c(2,3)])
  Fseries[[length(Fseries)+1]] <- m
}

names(Fseries) <- c(IDOto)

#cluster Bodden with cutoff test
clust_B <- tsclust(series = Bseries, type = "hierarchical",
                  k = 2L:6L,centroid = dba,
                  control = hierarchical_control(method = "ward.D"),
                  seed = 5L,args = tsclust_args(dist = list(window.size =5L)))

names(clust_B) <- paste0("k_",2L:6L)

#Internal cluster validity indexes to evaluate which number of clusters is ideal
EvalB <- data.frame(sapply(clust_B, cvi, type = "internal"))
#legend for validity index "max" -> index should be maximized; "min" -> index should be minimized
EvalB$win <- c("max", "max","max","min","min","max","min")

#cluster FW with cutoff test
clust_F <- tsclust(series = Fseries, type = "hierarchical",
                  k = 2L:6L,centroid = dba,
                  control = hierarchical_control(method = "ward.D"),
                  seed = 101L,args = tsclust_args(dist = list(window.size =5L)))

names(clust_F) <- paste0("k_",2L:6L)

#Internal cluster validity indexes to evaluate which number of clusters is ideal
EvalF <- data.frame(sapply(clust_B, cvi, type = "internal"))
#legend for validity index "max" -> index should be maximized; "min" -> index should be minimized
EvalF$win <- c("max", "max","max","min","min","max","min")

#Plot best solution lagoons
Lagoonplot <- plot(clust_B$k_4, type="sc", lwd=2)+
  ylim(-3,3)+ggtitle(expression("Sr/"~delta^18~O~"cluster centroids"))+
  labs(x="transect cell", y="scaled value")+
  theme(axis.text = element_text(size = 25, color="black"),
        axis.title = element_text(size = 25),
        plot.title = element_text(size = 25, hjust = 0.5))

#Plot best solution Freshwater
FWplot <- plot(clust_F$k_6, type="sc", linetype="solid", cex=2)+
  ylim(-3,3)+ggtitle(expression("Sr/"~delta^18~O~"cluster centroids"))+
  labs(x="transect cell", y="scaled value")+
  theme(axis.text = element_text(size = 25, color="black"),
        axis.title = element_text(size = 25),
        plot.title = element_text(size = 25, hjust = 0.5))

save(Fgroup,Bgroup,dataB, dataF, Bseries, Fseries, interpol, clust_B, clust_F, Lagoonplot, FWplot,
     file = "Data/Pike/Adults_hierarchical_clusters.RData")
```

```{r cluster assignment hierarchical, include=FALSE, eval=FALSE}
rm(list=ls())
load("Data/Pike/Adults_hierarchical_clusters.RData")
#assign clusters to fish
fishassign_B1 <- data.frame(fish_ID=names(Bseries), 
                            cluster=factor(clust_B$k_4@cluster, 
                                           levels = c(1:4), 
                                           labels = c("BW int saline|cold",
                                                      "BW Low saline|shift",
                                                      "BW high saline|warm",
                                                      "BW high saline|cold")))

fishassign_F1 <- data.frame(fish_ID=names(Fseries), 
                            cluster=factor(clust_F$k_6@cluster, 
                                           levels = c(1:6), 
                                           labels = c("FW low saline|shift",
                                                      "FW low saline|cold",
                                                      "FW int saline|const",
                                                      "FW high saline|shift",
                                                      "FW int saline|shift",
                                                      "FW high saline|const")))
#Join dtw clusters with pike data
fishassign_B2 <- Bgroup%>%
  group_by(fish_ID)%>%
  transmute(fish_ID=fish_ID,)%>%
  distinct()%>%
  inner_join(fishassign_B1, by="fish_ID")%>%
  mutate(cluster=fct_relevel(cluster,
                           "BW Low saline|shift",
                           "BW int saline|cold",
                           "BW high saline|warm",
                           "BW high saline|cold"))

fishassign_F2 <- Fgroup%>%
  group_by(fish_ID)%>%
  transmute(fish_ID=fish_ID,)%>%
  distinct()%>%
  inner_join(fishassign_F1, by="fish_ID")%>%
  mutate(cluster=fct_relevel(cluster,
                            "FW low saline|cold",
                            "FW low saline|shift",
                            "FW int saline|const",
                            "FW int saline|shift",
                            "FW high saline|const",
                            "FW high saline|shift"))

Groupall_clust <- rbind(fishassign_B2, fishassign_F2)

Groupall_transect <- rbind(Bgroup, Fgroup)

Groupall <- Groupall_transect%>%
  inner_join(Groupall_clust, by = c("fish_ID"))

fwrite(Groupall, "Data/Pike/clustering/Phenotypes-element-transects-all.csv")
```

```{r table of poportions, include=FALSE, eval=FALSE}
#Script to calculate simple distribution tables of phenotype number per area for the calculation of proportions
rm(list = ls())

Groupall <- fread("Data/Pike/clustering/Phenotypes-element-transects-all.csv")

allpike_pheno <- Groupall%>%
  mutate(phenotype=case_when(cluster=="FW low saline|cold"~"FW-resident",
                             cluster=="FW low saline|shift"~"FW-resident",
                             cluster=="FW int saline|const"~"FW-resident",
                             cluster=="FW int saline|shift"~"Anadromous",
                             cluster=="FW high saline|shift"~"Anadromous",
                             cluster=="FW high saline|const"~"Trans-habitat",
                             cluster=="BW Low saline|shift"~"Trans-habitat",
                             cluster=="BW int saline|cold"~"BW-resident",
                             cluster=="BW high saline|cold"~"BW-resident",
                             cluster=="BW high saline|warm"~"BW-resident"))%>%
  transmute(fish_ID=fish_ID, 
            area=area,
            phenotype=phenotype,
            cluster=cluster)%>%
  distinct()

Clust_props_per_area <- table(allpike_pheno$cluster)

Pheno_props_per_area <- table(allpike_pheno$area, allpike_pheno$phenotype)
write.table(Pheno_props_per_area, "Data/Pike/clustering/Lagoonprobs.csv", sep = ",")
```

```{r plot freshwater clusters, include=FALSE, eval=FALSE}
#Script to plot fine-grained cluster solution for freshwater sample (supplement)
rm(list=ls())

Groupall <- fread("Data/Pike/clustering/Phenotypes-element-transects-all.csv")

# Generate a color ramp between two colors
mypallette <- c(colorRampPalette(c("#C5D86D", "#00BFB2", "#F1C40F"))(6))

A <- ggplot(Groupall[Groupall$cluster=="FW low saline|cold",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-15,-10,10)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "black", size = 10),
        axis.ticks.x = element_line(colour = "transparent"),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.title.y = element_text(size = 10, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW low saline|cold (N = 4)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = -50, xmax = -50)+
  geom_segment(x=-450, y=-0.5, xend=-450, yend=8.5, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="darkorange")+
  annotation_custom(grob = text_grob(label = "increasing salinity", 
                                     rot = 90, size = 10),
                                     ymin=0, ymax=8,
                                     xmin=-460, xmax=-460)

A1 <- ggplot(Groupall[Groupall$cluster=="FW low saline|cold",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-15,-5,5.5)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "black", size = 8),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 10, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
    geom_segment(x=-450, y=0, xend=-450, yend=-3.2, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#FF6666")+
  annotation_custom(grob = text_grob(label = "warmer", 
                                     rot = 90,, size = 10),
                                     ymin=0, ymax=-3,
                                     xmin=-460, xmax=-460)+
  geom_segment(x=-450, y=0, xend=-450, yend=3, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#66CCCC")+
  annotation_custom(grob = text_grob(label = "colder", 
                                     rot = 90, size = 10),
                                     ymin=0, ymax=3.2,
                                     xmin=-460, xmax=-460)

B <- ggplot(Groupall[Groupall$cluster=="FW low saline|shift",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-5,-10,-6)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW low saline|shift (N = 9)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = 100, xmax = 100)

B1 <- ggplot(Groupall[Groupall$cluster=="FW low saline|shift",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-5,-5,-10)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

C <- ggplot(Groupall[Groupall$cluster=="FW int saline|const",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,5,-10,-16)), 
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW int saline|const (N = 14)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = 100, xmax = 100)

C1 <- ggplot(Groupall[Groupall$cluster=="FW int saline|const",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,5,-5,-20.5)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

D <- ggplot(Groupall[Groupall$cluster=="FW int saline|shift",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression(""))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,15,-10,-20)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.line = element_line(size = 0.3),
        axis.title.x = element_blank(),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW int saline|shift (N = 7)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = 100, xmax = 100)

D1 <- ggplot(Groupall[Groupall$cluster=="FW int saline|shift",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,15,-5,-31)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

E <- ggplot(Groupall[Groupall$cluster=="FW high saline|shift",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[5], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression(""))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,25,-10,-30)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.line = element_line(size = 0.3),
        axis.title.x = element_blank(),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW high saline|shift (N = 9)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = 100, xmax = 100)

E1 <- ggplot(Groupall[Groupall$cluster=="FW high saline|shift",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[5], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,25,-5,-41.5)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

F1 <- ggplot(Groupall[Groupall$cluster=="FW high saline|const",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[6], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression(""))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,35,-10,-40)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.line = element_line(size = 0.3),
        axis.title.x = element_blank(),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "FW high saline|const (N = 11)", size = 7, just = "left"),
                    ymin = 9, ymax = 9, xmin = 100, xmax = 100)

F2 <- ggplot(Groupall[Groupall$cluster=="FW high saline|const",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[6], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,35,-5,-51.5)),
        axis.text.x = element_text(color = "black", size = 8),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

Transects <- ggarrange(A, B, C, D, E, F1, A1, B1, C1, D1, E1, F2, labels = c("A", "B", "C", "D", "E", "F"), 
                       hjust = c(-3.85,-2.6,-1.2,0.2,1.6,3.4), vjust = c(2,2,2,2,2), 
                       font.label = list(size=10), nrow = 2, ncol = 6)

Transects2 <- annotate_figure(Transects, bottom = text_grob(expression("Distance from Core"~(mu~"m")), vjust = 1, size = 10))

ggsave("Figures/supplement/FW-clusters-life-history-transects.png", Transects2, dpi = 600, width = 24, height = 12, units = "cm", bg = "white")
```

```{r plot lagoon clusters, include=FALSE, eval=FALSE}
#Script to plot fine-grained cluster solution for freshwater sample (supplement)
rm(list=ls())

Groupall <- fread("Data/Pike/clustering/Phenotypes-element-transects-all.csv")

# Generate a color ramp between two colors
mypallette <- c(colorRampPalette(c("#004643", "#F1C40F"))(4))

A <- ggplot(Groupall[Groupall$cluster=="BW Low saline|shift",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-15,-10,10)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "black", size = 7),
        axis.ticks.x = element_line(colour = "transparent"),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "BW Low saline|shift (N = 20)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = -50, xmax = -50)+
  geom_segment(x=-450, y=-0.5, xend=-450, yend=8.5, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="darkorange")+
  annotation_custom(grob = text_grob(label = "increasing salinity", 
                                     rot = 90, size = 7),
                                     ymin=0, ymax=8,
                                     xmin=-460, xmax=-460)

A1 <- ggplot(Groupall[Groupall$cluster=="BW Low saline|shift",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-15,-5,5.5)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "black", size = 7),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
    geom_segment(x=-450, y=0, xend=-450, yend=-3.2, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#FF6666")+
  annotation_custom(grob = text_grob(label = "warmer", 
                                     rot = 90,, size = 7),
                                     ymin=0, ymax=-3,
                                     xmin=-460, xmax=-460)+
  geom_segment(x=-450, y=0, xend=-450, yend=3, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#66CCCC")+
  annotation_custom(grob = text_grob(label = "colder", 
                                     rot = 90, size = 7),
                                     ymin=0, ymax=3.2,
                                     xmin=-460, xmax=-460)

B <- ggplot(Groupall[Groupall$cluster=="BW int saline|cold",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-5,-10,-6)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "BW int saline|cold (N = 17)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

B1 <- ggplot(Groupall[Groupall$cluster=="BW int saline|cold",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-5,-5,-10)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

C <- ggplot(Groupall[Groupall$cluster=="BW high saline|warm",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,5,-10,-16)), 
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "BW high saline|warm (N = 22)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

C1 <- ggplot(Groupall[Groupall$cluster=="BW high saline|warm",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,5,-5,-20.5)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

D <- ggplot(Groupall[Groupall$cluster=="BW high saline|cold",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypallette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression(""))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,15,-10,-20)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.line = element_line(size = 0.3),
        axis.title.x = element_blank(),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "BW high saline|cold (N = 7)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

D1 <- ggplot(Groupall[Groupall$cluster=="BW high saline|cold",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypallette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,15,-5,-31)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

Transects <- ggarrange(A, B, C, D, A1, B1, C1, D1, labels = c("G", "H", "I", "J"), hjust = c(-3.85,-2.6,-1.2,0.15), vjust = c(2,2,2,2), font.label = list(size=10), nrow = 2, ncol = 4)

Transects2 <- annotate_figure(Transects, bottom = text_grob(expression("Distance from Core"~(mu~"m")), vjust = 1, size = 7))

ggsave("Figures/supplement/BW-clusters-life-history-transects.png", Transects2, dpi = 600, width = 16, height = 8, units = "cm", bg = "white")
```

```{r plot life history phenotypes, include=FALSE, eval=FALSE}
#Script to plot life history phenotypes identified by grouping ecologically similar clusters (behavioral phenotypes)
rm(list=ls())
Groupall <- fread("Data/Pike/clustering/Phenotypes-element-transects-all.csv")

mypalette <- adjustcolor(c("#004643","#00BFB2","#F1C40F","#C5D86D","#AF3800"), alpha.f = 0.6)

allpike_pheno <- Groupall%>%
  mutate(phenotype=case_when(cluster=="FW low saline|cold"~"FW-resident",
                              cluster=="FW low saline|shift"~"FW-resident",
                              cluster=="FW int saline|const"~"Anadromous",
                              cluster=="FW int saline|shift"~"Anadromous",
                              cluster=="FW high saline|shift"~"Anadromous",
                              cluster=="FW high saline|const"~"Trans-habitat",
                              cluster=="BW Low saline|shift"~"Trans-habitat",
                              cluster=="BW int saline|cold"~"BW-resident",
                              cluster=="BW high saline|cold"~"BW-resident",
                              cluster=="BW high saline|warm"~"BW-resident"))

A <- ggplot(allpike_pheno[allpike_pheno$phenotype=="FW-resident",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypalette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-15,-10,10)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "black", size = 7),
        axis.ticks.x = element_line(colour = "transparent"),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "freshwater resident (N = 27)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = -50, xmax = -50)+
  geom_segment(x=-450, y=-0.5, xend=-450, yend=8.5, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="darkorange")+
  annotation_custom(grob = text_grob(label = "increasing salinity", 
                                     rot = 90, size = 7),
                                     ymin=0, ymax=8,
                                     xmin=-460, xmax=-460)

A1 <- ggplot(allpike_pheno[allpike_pheno$phenotype=="FW-resident",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypalette[4], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-15,-5,5.5)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "black", size = 7),
        axis.ticks.y = element_line(size = 0.3),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3))+
  coord_cartesian(clip = "off")+
    geom_segment(x=-450, y=0, xend=-450, yend=-3.2, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#FF6666")+
  annotation_custom(grob = text_grob(label = "warmer", 
                                     rot = 90,, size = 7),
                                     ymin=0, ymax=-3,
                                     xmin=-460, xmax=-460)+
  geom_segment(x=-450, y=0, xend=-450, yend=3, 
               lineend = "butt", linejoin = "mitre",
               size=2, arrow = arrow(length = unit(0.18, "inches")),
               color="#66CCCC")+
  annotation_custom(grob = text_grob(label = "colder", 
                                     rot = 90, size = 7),
                                     ymin=0, ymax=3.2,
                                     xmin=-460, xmax=-460)

B <- ggplot(allpike_pheno[allpike_pheno$phenotype=="BW-resident",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypalette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,15,-10,-25.5)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "brackish resident (N = 46)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

B1 <- ggplot(allpike_pheno[allpike_pheno$phenotype=="BW-resident",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypalette[1], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,15,-5,-30)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

C <- ggplot(allpike_pheno[allpike_pheno$phenotype=="Anadromous",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypalette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression("Sr/Ca (mg/g)"))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,-5,-10,-6)), 
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "anadromous type (N = 16)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

C1 <- ggplot(allpike_pheno[allpike_pheno$phenotype=="Anadromous",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypalette[2], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,-5,-5,-10)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")

D <- ggplot(allpike_pheno[allpike_pheno$phenotype=="Trans-habitat",])+
  geom_line(aes(dist, Sr*1000, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, Sr*1000), color="black", fill=mypalette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(0, 8)+
  scale_y_continuous(breaks = seq(0,8,1), limits = c(0,8))+
  xlim(0,2100)+
  labs(y=expression(""))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(20,5,-10,-9.5)),
        axis.text.x = element_text(color = "transparent", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3, colour = "transparent"),
        axis.title.y = element_text(size = 7, margin = margin(r=10)),
        axis.line = element_line(size = 0.3),
        axis.title.x = element_blank(),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")+
  annotation_custom(grob = text_grob(label = "cross-habitat type (N = 31)", size = 7, just = "left"),
                    ymin = 9.3, ymax = 9.3, xmin = 100, xmax = 100)

D1 <- ggplot(allpike_pheno[allpike_pheno$phenotype=="Trans-habitat",])+
  geom_line(aes(dist, res, group=fish_ID), color="grey", linewidth=.3, alpha=.8)+
  geom_smooth(aes(dist, res), color="black", fill=mypalette[3], 
              alpha=1, size=.3, method = "gam")+
  theme_classic()+
  ylim(-3, 3)+
  xlim(0,2100)+
  labs(y=expression("scaled"~delta^18*"O"~("VPDB \u2030")))+
  theme(legend.position = "none", 
        panel.grid.major.y = element_line(size = .3),
        panel.border = element_rect(fill = NA, size = .3),
        plot.margin = margin(c(10,5,-5,-20)),
        axis.text.x = element_text(color = "black", size = 7),
        axis.text.y = element_text(color = "transparent", size = 7),
        axis.ticks.y = element_line(size = 0.3, colour = "transparent"),
        axis.ticks.length.y = unit(0.05, "cm"),
        axis.ticks.x = element_line(size = 0.3),
        axis.title.y = element_text(size = 7, margin = margin(r=10), colour = "transparent"),
        axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        plot.background = element_blank())+
  coord_cartesian(clip = "off")


Transects <- ggarrange(A, C, D, B, A1, C1, D1, B1, labels = c("A", "B", "C", "D"), hjust = c(-3.85,-2.6,-1.2,0.15), vjust = c(2,2,2,2), font.label = list(size=10), nrow = 2, ncol = 4)

Transects2 <- annotate_figure(Transects, bottom = text_grob(expression("Distance from Core"~(mu~"m")), vjust = 1, size = 7))

ggsave("Figures/MS/Ecotypes-life-history-transects.png", Transects2, dpi = 600, width = 16, height = 8, units = "cm", bg = "white")
```

```{r dissimilarity test, include=FALSE, eval=FALSE}
#Script to perform dissimilarity test (PERMANOVA) and jackknife reclassification on clusters
rm(list = ls())

#read in data
Groupall <- fread("Data/Pike/clustering/Phenotypes-element-transects-all.csv")

dataLagoon <- Groupall%>%
  filter(area%in%c("WRB", "NRB", "GB"))%>%
  transmute(fish_ID=fish_ID,
            Sr=Sr,
            d18O_res=res,
            year=year,
            cluster=cluster)

dataFW <- Groupall%>%
  filter(!area%in%c("WRB", "NRB", "GB"))%>%
  transmute(fish_ID=fish_ID,
            Sr=Sr,
            d18O_res=res,
            year=year,
            cluster=cluster)

#transform data to all positive
dataLagoon$d18O_res <- (dataLagoon$d18O_res+sqrt((min(dataLagoon$d18O_res))^2))
#So a value of d18O res is now to be interpreted as residual from predicted
#value + 2.518462

dataLagoon$Sr <- (dataLagoon$Sr+sqrt((min(dataLagoon$Sr))^2))
#So a value of Sr is now to be interpreted as scaled value + 2.716513

#Same for reshwater group
dataFW$d18O_res <- (dataFW$d18O_res+sqrt((min(dataFW$d18O_res))^2))
dataFW$Sr <- (dataFW$Sr+sqrt((min(dataFW$Sr))^2))

#join
data <- rbind(dataLagoon, dataFW)

#make list of matrices Bodden
IDOto <- unique(data$fish_ID)
series <- list()

for (i in 1:length(IDOto)) {
  m <- as.matrix(data[which(data$fish_ID==IDOto[i]),])
  series[[length(series)+1]] <- m
}

names(series) <- c(IDOto)

#Revert from list to data frame, assign cell numbers
Series <- do.call(rbind.data.frame,series)
Series$ID <- substr(rownames(Series), 1,8)
rownames(Series) <- NULL

#Assign years based on cell numbers, join with cluster ID
Series1 <- Series%>%
  transmute(fish_ID=fish_ID,
            year=case_when(year<2~"orig",
                           year>1&year<3~"juv",
                           year>2~"adult"),
            cluster=cluster,
            Sr=Sr,
            d18O_res=d18O_res)%>%
  group_by(fish_ID, year)%>%
  summarize(fish_ID=fish_ID,
            year=year,
            cluster=cluster,
            Sr=mean(as.numeric(Sr)),
            d18O_res=mean(as.numeric(d18O_res)))%>%
  distinct()

#Make wide for testing
Series2 <- pivot_wider(Series1, names_from = year, values_from = c(Sr, d18O_res))

#cluster needs to be a factor
Series2$cluster <- as.factor(Series2$cluster)

#make mvabund object of Sr and d18O data
elements <- mvabund(Series2[,3:8])

Pikedist <- vegdist(elements, method = "euclidean", na.rm = T)
Disp <- betadisper(Pikedist, Series2$cluster)
anova(Disp)

plot(Disp)

adonis2(elements~Series2$cluster, method = "euclidean", permutations = 9999, na.rm=T)

set.seed(128)

Series3 <- na.omit(Series2)

Series3 <- Series3%>%
  mutate(cluster=case_when(cluster=="FW low saline|cold"~"FW_resident",
                            cluster=="FW low saline|shift"~"FW_resident",
                            cluster=="FW int saline|const"~"Anadromous",
                            cluster=="FW int saline|shift"~"Anadromous",
                            cluster=="FW high saline|shift"~"Anadromous",
                            cluster=="FW high saline|const"~"Trans-habitat",
                            cluster=="BW Low saline|shift"~"Trans-habitat",
                            cluster=="BW int saline|cold"~"BW-resident",
                            cluster=="BW high saline|cold"~"BW-resident",
                            cluster=="BW high saline|warm"~"BW-resident"))

#jackknife reclassification
lda.m2 <- lda(cluster~Sr_orig+Sr_juv+Sr_adult+d18O_res_orig+d18O_res_juv+d18O_res_adult, 
              data = Series3, CV=TRUE)

Series3$jacknife <- lda.m2$class
summary <- Series3 %>% dplyr::summarise(cluster=cluster,
                                        jackknife=jacknife,
                                        lda.error=mean(cluster != jacknife))
```

# Genotyping individual pike
**This section only describes how the output of the individual-level genotyping detailed in section D of the supplementary material was processed for growth analysis and phenotype-genotype matching. For details/code on the analysis process of the SNP panel development, GT-sequencing, STRUCTURE analysis etc., contact Stefan Dennenmoser (stefan.dennenmoser@uni-oldenburg.de), working group ecological genomics, Institute of Biology and Environmental Sciences, Carl von Ossietzky Universität Oldenburg, Carl von Ossietzky-Str. 9-11, 26111 Oldenburg, Germany**

##Matching phenotype to genotype
```{r clean up genotype data, include=FALSE, eval=FALSE}
#Script to combine phenotype data with genotype assignment probabilities
rm(list = ls())

substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

#Load STRUCTURE output file
geneticsdata <- read.delim("Data/pike/genetics/struct_K3_1_f_clean.txt", sep = "", dec = ".", header = F, stringsAsFactors = F)

#Load phenotype data
Otodata <- fread("Data/pike/growth/Pikedata_model.csv")

#Gentypes: 1=Freshwater; 2=brackish; 3=anadromous
Data <- geneticsdata%>%
  transmute(ID=paste0(sub("_","-",substrRight(V2,8))),
            cluster1=as.numeric(V6),
            cluster2=as.numeric(V7),
            cluster3=as.numeric(V8))%>%
  drop_na()

Otoclusters <- Otodata%>%
  transmute(ID=ID,
            DTW_cluster=cluster,
            phenotype=phenotype,
            waterbody=waterbody)%>%
  distinct()

Otogen <- Data%>%left_join(Otoclusters, by="ID")%>%
  drop_na()

fwrite(Otogen, "Data/pike/genetics/Phenotype-genotype-three-clusters.csv")

# Reshape the data frame to long format
df_long <- pivot_longer(Otogen, -c(ID, phenotype, DTW_cluster, waterbody), names_to = "cluster", values_to = "prob")

# Filter data for ecotype "A"
df_long2 <- df_long%>%
  transmute(phenotype=phenotype,
            ID=ID,
            DTW_cluster=DTW_cluster,
            waterbody=waterbody,
            cluster=case_when(cluster=="cluster3"~"anadromous",
                              cluster=="cluster2"~"brackish",
                              cluster=="cluster1"~"freshwater"),
            ass_prob=prob)

fwrite(df_long2, "Data/pike/genetics/Genotype-phenotype-combined.csv")
```

```{r hard cluster assignment, include=FALSE, eval=FALSE}
#Script to combine phenotype data with discrete genotype based on 0.7 probability treshold
rm(list = ls())

substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

#Load STRUCTURE output file
geneticsdata <- read.delim("Data/pike/genetics/struct_K3_1_f_clean.txt", sep = "", dec = ".", header = F, stringsAsFactors = F)

#Load phenotype data
Otodata <- fread("Data/pike/growth/Pikedata_model.csv")

#Genotypes: 1=Freshwater; 2=brackish; 3=anadromous
Data <- geneticsdata%>%
  transmute(ID=paste0(sub("_","-",substrRight(V2,8))),
            cluster1=as.numeric(V6),
            cluster2=as.numeric(V7),
            cluster3=as.numeric(V8))%>%
  drop_na()

Otoclusters <- Otodata%>%
  transmute(ID=ID,
            DTW_cluster=cluster,
            phenotype=phenotype,
            waterbody=waterbody)%>%
  distinct()

Otogen <- Data%>%left_join(Otoclusters, by="ID")%>%
  drop_na()

Otogen$clust <- apply(Otogen[,c(2,3,4)], 1, FUN = which.max)
Otogen$clustval <- apply(Otogen[,c(2,3,4)], 1, FUN = max)

Otogen2 <- Otogen%>%
  mutate(hardclust = case_when(clustval < 0.7 ~ "U",
                            TRUE~as.character(clust)))

Otogen2$hardclust <- fct_relevel(Otogen2$hardclust, "1", "3", "2", "U")

fwrite(Otogen2, file = "Data/pike/genetics/discrete-cluster-assignment.csv")
```

# Growth analysis of life history phenotypes and genotype
## Age-specific growth
```{r get clean increment growth data, include=FALSE, eval=FALSE}
#Script combine age readings on pike otoliths from all samples into one data frame and join with individual metadata
rm(list = ls())

#read in pike data
Pike1 <- fread("Data/pike/1Pike_Latest_Version.txt")
Pike2 <- Pike1%>%
  transmute(ID=case_when(Fish_ID=="BH-91050.2"~"BH-91050",
                         Fish_ID=="BH-51614"&Waterbody=="BAT"~"NA",
                         Fish_ID=="BH-90307"&MarkorRecap=="Mark"~"NA",
                         TRUE~Fish_ID),
            capt_date=Date,
            TL_mm = Total_Length_mm,
            Weight=Weight,
            sex=Sex,
            area=Waterbody)

#load in age readings
pike_increments_2022 <- read.delim("Data/pike/growth/Sample_2022_oto_increments.txt", header = T, stringsAsFactors = F)
pike_increments_2021 <- read.delim("Data/pike/growth/Sample_2021_oto_increments.txt", header = T, stringsAsFactors = F)
pike_increments_pilot <- read.delim("Data/pike/growth/Pilot_sample_oto_increments.txt", header = T, stringsAsFactors = F)

pike_increments_pilot$Sample <- substr(pike_increments_pilot$Sample,3,10)

pike_increments <- rbind(pike_increments_2021, pike_increments_2022, pike_increments_pilot)

substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

pike_increments2 <- pike_increments%>%
  group_by(Sample)%>%
  transmute(ID=substrRight(Sample,8),
            Quality=Quality,
            Age=Age,
            Cohort=Cohort,
            Core=Core,
            lifeyear=I,
            total_rad_um=sum(Increment, na.rm = T),
            year=Year,
            Increment=Increment)%>%
  ungroup()

pike_inc <- pike_increments2 %>% inner_join(Pike2, by="ID")%>%
  group_by(ID)%>%
  transmute(ID=ID,
            age=Age,
            cohort=Cohort,
            capt=area,
            TL=TL_mm,
            weigth=Weight,
            sex=sex,
            date=lubridate::date(parse_date_time(capt_date, "ymd")),
            core=Core,
            lifeyear=as.integer(lifeyear),
            year=year,
            increment=Increment,
            total_rad_um=total_rad_um,
            quality=Quality)%>%
  drop_na(lifeyear)%>%
  arrange(ID,lifeyear)

pike_inc <- as.data.frame(pike_inc)

#convert inc to rad
IDpike <- unique(pike_inc$ID)
rad <- NULL
for (i in 1:length(IDpike)) {
  d <- cumsum(coalesce(pike_inc[which(pike_inc$ID == IDpike[i]), "increment"], 0)) + pike_inc[which(pike_inc$ID == IDpike[i]), "increment"]*0
  
  rad <- c(rad, d)
}

pike_inc$rad <- rad

pike_inc2 <- pike_inc%>%
  group_by(ID)%>%
  transmute(ID=ID,
            age=age,
            cohort=cohort,
            capt=capt,
            TL_mm=TL,
            weight_g=weigth,
            sex=sex,
            date=date,
            core=core,
            lifeyear=lifeyear,
            year=year,
            increment_um=increment,
            rad_um=rad,
            total_rad_um=total_rad_um,
            quality=quality)%>%
  drop_na(lifeyear)

fwrite(pike_inc2, "Data/pike/growth/Oto_increment_data_with_rad.csv")
```

```{r join growth measurements with behavioral phenotypes, include=FALSE, eval=FALSE}
#Script to join element transects, age and increment data with life history phenotypes
rm(list = ls())

#load phenotype data
allpike_clust <- fread("Data/pike/clustering/Phenotypes-element-transects-all.csv")

#load element interpolated element data (years not smoothed over for later means)
oto_elements <- fread("Data/pike/elements/Oto_LA-to-d18O_interpolation_Sr-corrected.csv")

#load increment data
pike_inc <- fread("Data/pike/growth/Oto_increment_data_with_rad.csv")

#join increment data with clustering results
allpike_clust2 <- allpike_clust%>%
  transmute(ID=fish_ID,
            cluster=cluster,
            phenotype=case_when(cluster=="FW low saline|cold"~"FW_resident",
                              cluster=="FW low saline|shift"~"FW_resident",
                              cluster=="FW int saline|const"~"FW_resident",
                              cluster=="FW int saline|shift"~"Anadromous",
                              cluster=="FW high saline|shift"~"Anadromous",
                              cluster=="FW high saline|const"~"Trans-habitat",
                              cluster=="BW Low saline|shift"~"Trans-habitat",
                              cluster=="BW int saline|cold"~"BW-resident",
                              cluster=="BW high saline|cold"~"BW-resident",
                              cluster=="BW high saline|warm"~"BW-resident"))

pike_years <- oto_elements%>%
  group_by(fish_ID,lifeyear)%>%
  summarise(ID=fish_ID,
            lifeyear=as.numeric(lifeyear),
            res_mean=mean(res, na.rm=T),
            Sr_mean=mean(Sr, na.rm=T))%>%
  drop_na()

pikedata <- pike_inc%>%
  inner_join(pike_years, by = c("ID", "lifeyear"))%>%
  inner_join(allpike_clust2, by = c("ID"))%>%
  mutate(area = case_when(capt=="SB"~"WRB",
                           capt=="KB"~"WRB",
                           capt=="Sehrowbach"~"Tributary",
                           capt=="Ziese"~"Tributary",
                           capt=="NHG"~"Tributary",
                           capt=="WB"~"NRB",
                           capt=="BEG"~"NRB",
                           capt=="KJB"~"NRB",
                           capt=="GJB"~"NRB",
                           capt=="GB"~"GB",
                           capt=="Peene"~"Tributary",
                           capt=="Barthe"~"Tributary",
                           capt=="Badendycksgraben"~"Tributary"))%>%
  distinct()

fwrite(pikedata, "Data/pike/growth/Growthdata_element_clusters.csv")
```

```{r join phenotype & genotype & cleanup, include=FALSE, eval=FALSE}
# cleanup script for GLMM modelling joining phenotype and genotype data
rm(list = ls())

pikedata <- fread("Data/pike/growth/Growthdata_element_clusters.csv")

#Inspect data
head(pikedata)
tail(pikedata)
str(pikedata)

#Prepare data frame with factors, scale and center element data
pikes2 <- pikedata%>%
  transmute(ID=ID,
            waterbody=as.factor(area),
            TL_mm=TL_mm,
            cluster=as.factor(cluster),
            phenotype=as.factor(phenotype),
            sex=as.factor(sex),
            lifeyear=lifeyear,
            inc_um=increment_um,
            rad_um=rad_um,
            res_mean=as.numeric(scale(res_mean)),
            Sr_mean=as.numeric(scale(Sr_mean)))%>%
  drop_na(res_mean, Sr_mean)

pikes2[pikes2$ID=="BH-90307",]$sex <- "f"

#Join genotype data
genotypes <- fread("Data/pike/genetics/discrete-cluster-assignment.csv")

genotype_only <- genotypes%>%
  transmute(ID=ID,
            genotype=hardclust)

pikes3 <- pikes2%>%
  left_join(genotype_only, by = "ID")

fwrite(pikes3, "Data/pike/growth/Pikedata_model.csv")

#Check
str(pikes3)
```

```{r data exploration, include=FALSE, echo=FALSE, eval=FALSE, fig.height=6, fig.width=8}
rm(list = ls())

#read in model data
pikes <- fread("Data/pike/growth/Pikedata_model.csv")

#Exploration script
par(mfrow=c(2,2))
dotchart(pikes2$inc_um, color = as.factor(pikes2$phenotype), main = "Color = Phenotype(Cluster)")
dotchart(pikes2$inc_um, color = as.factor(pikes2$genotype), main = "Color = Genotype(STRUCTURE Cluster)")
dotchart(pikes2$inc, color = as.factor(pikes2$sex), main = "Color = Sex")
dotchart(pikes2$inc, color = as.factor(pikes2$lifeyear), main = "Color = d18O")

par(mfrow=c(3,2))
boxplot(pikes2$inc~pikes2$lifeyear, ylab ="Growth increment", xlab = "Age")
boxplot(pikes2$inc~pikes2$phenotype, ylab ="Growth increment", xlab = "Phenotype")
boxplot(pikes2$inc~pikes2$genotype, ylab ="Growth increment", xlab = "Genotype")
boxplot(pikes2$inc~pikes2$sex, ylab ="Growth increment", xlab = "Sex")
plot(pikes2$inc~pikes2$res_mean, ylab ="Growth increment", xlab = "d18O")
plot(pikes2$inc~pikes2$Sr_mean, ylab ="Growth increment", xlab = "Sr")
plot(0, xaxt = 'n', yaxt = 'n', bty = 'n', pch = '', ylab = '', xlab = '')

par(mfrow = c(2,2))
hist(pikes2$lifeyear^2, main = "", xlab = "Increment (um)")
hist(pikes2$res_mean, main = "", xlab = "d18O ratio")
hist(pikes2$Sr_mean, main = "", xlab = "Sr concentration")
plot(0, xaxt = 'n', yaxt = 'n', bty = 'n', pch = '', ylab = '', xlab = '')
```

```{r decide on age effect, include=FALSE, eval=FALSE}
# Check which allometric scaling expoonent works best on age
rm(list = ls())

#read in model data
pikes <- fread("Data/pike/growth/Pikedata_model.csv")

#Run model
pikemodel.1 <- lmer(log10(inc_um)~lifeyear+exp(-lifeyear)+sex+res_mean+Sr_mean+lifeyear*phenotype+lifeyear*genotype+(1|ID),
                    REML = F, data = pikes)
pikemodel.1.1 <- lmer(log10(inc_um)~lifeyear+I(lifeyear^2)+sex+res_mean+Sr_mean+lifeyear*phenotype+lifeyear*genotype+(1|ID),
                    REML = F, data = pikes)

anova(pikemodel.1, pikemodel.1.1)
#Quadratic term has lower AIC and residuals look better

summary(pikemodel.1)

#Check variance inflation factors
pike.vif <- vif(pikemodel.1.1)

barplot(pike.vif[,3], main = "VIF values", horiz = F, col = "darkgreen", ylim = c(0,10),
        las = 2)
abline(h = 5, lwd = 3, lty = 2, col="red")
# Lifeyear (age) crosses a VIF of 5, as it is obviously autocorrelated with age^2, which we deemed acceptable in the context
# of our model, as we aimed only at accounting for allometry, not evaluate the effect of age directly
```

```{r test effects, echo=TRUE, eval=FALSE}
rm(list = ls())

#read in model data
pikes <- fread("Data/pike/growth/Pikedata_model.csv")

#run model
pikemodel.1<- lmer(log10(inc_um)~res_mean+lifeyear*phenotype+lifeyear*genotype+I(lifeyear^2)+Sr_mean+sex+(1|ID),
                    REML = F, data = pikes)

#Test random effects
rand(pikemodel.1)
#random effect is significant

#Test age effect
pikemodel.noage <- update(pikemodel.1, ~.-I(lifeyear^2))

anova(pikemodel.1, pikemodel.noage)
#age is significant

#test genotype effect
pikemodel.nogen <- update(pikemodel.1, ~.-lifeyear*genotype)

anova(pikemodel.1, pikemodel.nogen)

#New model with genotype removed
pikemodel.1.1 <- lmer(log10(inc_um)~res_mean+lifeyear*phenotype+I(lifeyear^2)+Sr_mean+sex+(1|ID),
                    REML = F, data = pikes)

#Test random effects
rand(pikemodel.1.1)
#random effect is significant

#Test age effect
pikemodel.noage <- update(pikemodel.1.1, ~.-I(lifeyear^2))

anova(pikemodel.1.1, pikemodel.noage)

#Test life history phenotype effect (interaction term)
pikemodel.nopheno <- update(pikemodel.1.1, ~.-lifeyear*phenotype)

anova(pikemodel.1.1, pikemodel.nopheno)
#age*phenotype is significant

#Test thermal effects
pikemodel.nores <- update(pikemodel.1.1, ~.-res_mean)

anova(pikemodel.1.1, pikemodel.nores)
#relative d18O (thermal proxy) significant

#Test salinity effects
pikemodel.noSr <- update(pikemodel.1.1, ~.-Sr_mean)

anova(pikemodel.1.1, pikemodel.noSr)
#Salinity (Sr:Ca) significant

#Test sex effects
pikemodel.nosex <- update(pikemodel.1.1, ~.-sex)

anova(pikemodel.1.1, pikemodel.nosex)
#sex is significant
```

```{r age-specific model summary and table, out.width="120%", dpi=300, echo=FALSE, eval=FALSE, message=FALSE}
rm(list = ls())

#read in model data
pikes <- fread("Data/pike/growth/Pikedata_model.csv")

pikemodel.final <- lmer(log10(inc_um)~res_mean+lifeyear*phenotype+I(lifeyear^2)+Sr_mean+sex+(1|ID),
                    REML = F, data = pikes)

summary(pikemodel.final)
sd(resid(pikemodel.final))
r.squaredGLMM(pikemodel.final)
tab_model(pikemodel.final, show.fstat = T, show.loglik = T, digits.re = 20, show.re.var = T)
```

```{r age-specific model validation, echo=FALSE, eval=FALSE}
rm(list = ls())

#read in model data
pikes <- fread("Data/pike/growth/Pikedata_model.csv")

#validation
pikemodel.final <- lmer(log10(inc_um)~res_mean+lifeyear*phenotype+I(lifeyear^2)+Sr_mean+sex+(1|ID),
                    REML = F, data = pikes)

pikes$resid <- resid(pikemodel.final)

png(filename = "Figures/supplement/S2_residual_plot.png", width = 5000, height = 2500, res = 600)
plot(pikemodel.final, xlab = "Fitted", ylab = "Residuals")
dev.off()

png(filename = "Figures/supplement/S3_residual_hist.png", width = 5000, height = 2500, res = 600)
hist(pikes2$resid, main = "", xlab = "Residuals")
dev.off()

png(filename = "Figures/supplement/S4_qqnorm.png", width = 5000, height = 2500, res = 600)
qqnorm(resid(pikemodel.final))
qqline(resid(pikemodel.final))
dev.off()

png(filename = "Figures/supplement/S5_params.png", width = 5000, height = 2500, res = 600)
par(mfrow=c(2,3))
plot(pikes2$resid~as.factor(pikes2$phenotype), ylab = "Residuals", xlab = "", las = 2)
plot(pikes2$resid~as.factor(pikes2$sex), ylab = "Residuals", xlab = "Sex")
plot(pikes2$resid~pikes2$lifeyear, ylab = "Residuals", xlab = "Age")
plot(pikes2$resid~pikes2$res_mean, ylab = "Residuals", xlab = "d18O")
plot(pikes2$resid~pikes2$Sr_mean, ylab = "Residuals", xlab = "Sr concentration")
dev.off()
```

```{r figures for abiotic main effects, eval=FALSE}
rm(list = ls())

pikes <- fread("Data/pike/growth/Pikedata_model.csv")

#factorize temperature
pikes$agefact <- ifelse(pikes$lifeyear<3, 2, 
                         ifelse(pikes$lifeyear>2&pikes$lifeyear<7, 1,
                                ifelse(pikes$lifeyear>6, 3, NA)))

pikes$agefact2 <- factor(pikes$agefact,levels = c(2,1,3),
                                            labels = c("0-2 years", "3-6 years", 
                                                       "> 6 years"))

mypallette <- viridis::viridis(4, direction = -1)

#Temperature effect
Temp <- ggplot(pikes, aes(res_mean, log(inc_um), color=agefact2))+
  geom_point(size=.5)+
  stat_smooth(method = "lm", fill="gray", fullrange = T, lwd=.8)+
  labs(x=expression("scaled"~delta^18*"O"~("z-score")), 
       y=expression("log increment"~(mu*"m")),
       color="ageclass")+
  scale_color_viridis_d()+
  theme_classic()+
  theme(panel.grid.major = element_line(),
        legend.position = c(.9,.82),
        legend.background = element_blank(),
        legend.box.background = element_rect(),
        axis.line = element_line(size = .5, colour = "black", linetype = 1),
        axis.ticks = element_line(size = .5, colour = "black"),
        axis.text = element_text(size = 15, colour = "black"),
        axis.title.x = element_text(size = 15, margin = margin(t=15), colour = "black"),
        axis.title.y = element_text(size = 15, colour = "black"),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 15, colour = "black"),
        legend.key.size = unit(.8,"line"))+
  coord_cartesian(clip = "off")+
  geom_segment(x=0, y=3.2, xend=2.5, yend=3.2, 
               lineend = "butt", linejoin = "mitre",
               size=3, arrow = arrow(length = unit(0.18, "inches")),
               color="#66CCCC")+
  annotation_custom(grob = text_grob(label = "colder", 
                                     rot = 0, face = "plain", size = 13),
                                     ymin=3.22, ymax=3.22,
                                     xmin=0, xmax=2.5)+
  geom_segment(x=0, y=3.2, xend=-2.5, yend=3.2, 
               lineend = "butt", linejoin = "mitre",
               size=3, arrow = arrow(length = unit(0.18, "inches")),
               color="#FF6666")+
  annotation_custom(grob = text_grob(label = "warmer", 
                                     rot = -0, face = "plain", size = 13),
                                     ymin=3.22, ymax=3.22,
                                     xmin=0, xmax=-2.5)

Sal <- ggplot(pikes, aes(Sr_mean, log(inc_um), color=agefact2))+
  geom_point(size=.5)+
  stat_smooth(method = "lm", fill="gray", fullrange = T, lwd=.8)+
  labs(x=expression("scaled"~"Sr/Ca"~("z-score")), 
       y=expression("log increment"~(mu*"m")),
       color="ageclass")+
  scale_color_viridis_d()+
  theme_classic()+
  theme(panel.grid.major = element_line(),
        legend.position = c(.9,.92),
        legend.background = element_blank(),
        legend.box.background = element_rect(),
        axis.line = element_line(size = .5, colour = "black", linetype = 1),
        axis.ticks = element_line(size = .5, colour = "black"),
        axis.text = element_text(size = 15, colour = "black"),
        axis.title.x = element_text(size = 15, margin = margin(t=15), colour = "black"),
        axis.title.y = element_text(size = 15, colour = "black"),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 15, colour = "black"),
        legend.key.size = unit(.8,"line"),
        plot.margin = margin(25,1,1,1))+
  coord_cartesian(clip = "off")+
  geom_segment(x=0, y=3.2, xend=2, yend=3.2, 
               lineend = "butt", linejoin = "mitre",
               size=3, arrow = arrow(length = unit(0.18, "inches")),
               color="darkorange")+
  annotation_custom(grob = text_grob(label = "higher salinity", 
                                     rot = 0, face = "plain", size = 13),
                                     ymin=3.22, ymax=3.22,
                                     xmin= 0.5, xmax=1.5)+
  geom_segment(x=0, y=3.2, xend=-2, yend=3.2, 
               lineend = "butt", linejoin = "mitre",
               size=3, arrow = arrow(length = unit(0.18, "inches")),
               color="lightblue")+
  annotation_custom(grob = text_grob(label = "lower salinity", 
                                     rot = -0, face = "plain", size = 13),
                                     ymin=3.22, ymax=3.22,
                                     xmin= -0.5, xmax= -1.5)

Tempsal <- ggarrange(Temp, Sal, nrow = 1, labels = c("A", "B"), common.legend = T)

ggsave("Figures/MS/Temperature-Salinity-effect-growth.png", Tempsal,
       width = 16, height = 9, units = "cm", dpi = 600, bg = "white")
```

```{r figure for phenotye effect, eval=TRUE}
rm(list = ls())

pikes <- fread("Data/pike/growth/Pikedata_model.csv")

pikemodel.final <- lmer(log10(inc_um)~res_mean+lifeyear*phenotype+I(lifeyear^2)+Sr_mean+sex+(1|ID),
                    REML = F, data = pikes)

mypalette1 <- adjustcolor(c("#004643","#00BFB2","#F1C40F","#C5D86D","#AF3800"), alpha.f = 0.6)

lifeyear <- unique(pikes$lifeyear)
phenotype <- unique(pikes$phenotype)
genotype <- unique(pikes$genotype)
res_mean <- seq(min(pikes$res_mean), max(pikes$res_mean), 0.1)
Sr_mean <- seq(min(pikes$Sr_mean), max(pikes$res_mean),0.1)
sex <- unique(pikes$sex)
ID <- "NA"

Newdata <- expand.grid(lifeyear=lifeyear, phenotype=phenotype, genotype=genotype, res_mean=res_mean, Sr_mean=Sr_mean, sex=sex, ID=ID)
head(Newdata)

Newdata$prediction <- 10^(predict(pikemodel.final, newdata = Newdata, allow.new.levels=TRUE, type="response"))

Newdata2 <- Newdata%>%
  mutate(keep=case_when(phenotype=="FW_resident"&lifeyear<=max(pikes[pikes$phenotype=="FW_resident",]$lifeyear)~1,
                   phenotype=="Anadromous"&lifeyear<=max(pikes[pikes$phenotype=="Anadromous",]$lifeyear)~1,
                   phenotype=="BW-resident"&lifeyear<=max(pikes[pikes$phenotype=="BW-resident",]$lifeyear)~1,
                   phenotype=="Trans-habitat"&lifeyear<=max(pikes[pikes$phenotype=="Trans-habitat",]$lifeyear)~1,
                   TRUE~0))%>%
           filter(keep==1)

Newdata2$phenotype <- fct_relevel(Newdata2$phenotype, "FW_resident", "Anadromous", "Trans-habitat", "BW-resident")

# Create ggplot with boxplots and custom position dodge 

phenogrowth <- ggplot(Newdata2, aes(factor(lifeyear), prediction, fill=phenotype), alpha=0.6)+
  geom_boxplot(lwd=.5, outlier.shape = NA, width=.6, position = position_dodge(width = NULL), color="black")+
  labs(x="Age (years)", y=expression("predicted increment"~(mu*"m")),
       fill="behavioral phenotype")+
  scale_x_discrete(limits=factor(c(1:11)),
                   breaks=c("1","2","3","4","5","6","7","8","9","10","11"))+
  scale_fill_manual(breaks = c("FW_resident","Anadromous","Trans-habitat","BW-resident"),
                    values = c("#C5D86D","#00BFB2","#F1C40F","#004643"),
                    labels = c("freshwater resident", "anadromous","cross-habitat", "brackish resident"))+
  theme_classic()+
  theme(panel.grid.major = element_line(),
        legend.position = c(.715,.79),
        legend.background = element_blank(),
        legend.box.background = element_rect(),
        axis.line = element_line(size = 1, colour = "black", linetype = 1),
        axis.ticks = element_line(size = 1, colour = "black"),
        axis.text = element_text(size = 30, colour = "black"),
        axis.text.y = element_text(size = 30, colour = "black"),
        axis.title.x = element_text(size = 30, colour = "black"),
        axis.title.y = element_text(size = 30, colour = "black"),
        legend.text = element_text(size = 30, colour = "black"),
        legend.title = element_text(size = 30, colour = "black"),
        legend.key.size = unit(2.5,"line"))

ggsave("Figures/MS/Phenotype-effects-growth.png", phenogrowth, width = 16, height = 8, dpi = 600, bg = "white")
```

## Lifelong growth
```{r data prep for VB modelling, include=FALSE, eval=FALSE}
rm(list = ls())

# load data
pikedata <- fread("Data/pike/growth/Pikedata_model.csv")

# Cleanup
growthdata <- pikedata %>%
  transmute(ID = ID,
            ID2 = as.numeric(factor(ID)),
            sex = sex,
            TL_mm = TL_mm,
            rad_mm = rad_um/1000,
            waterbody = waterbody,
            lifeyear = lifeyear,
            phenotype_char = case_when(phenotype == "FW_resident"~"FW-resident",
                                   TRUE~phenotype),
            phenotype_num = case_when(phenotype == "FW_resident" ~ 1,
                                  phenotype == "Anadromous" ~ 2,
                                  phenotype == "Trans-habitat" ~ 3,
                                  phenotype == "BW-resident" ~ 4),
            genotype_num = case_when(genotype == "1" ~ 1,
                                     genotype == "2" ~ 2,
                                     genotype == "3" ~ 3,
                                     genotype == "U" ~ 4))

fwrite(growthdata, file = "Data/Pike/growth/Pikedata_Stan.csv")
```

```{r run phenotype model and save output, include=FALSE, eval=FALSE}
rm(list = ls())

source("Data/pike/growth/funcs.r")
growthdata <- fread("Data/Pike/growth/Pikedata_Stan.csv")


standata_growth = list(
	Radmm = growthdata$rad_mm, 
	Agei = growthdata$lifeyear,
	N = nrow(growthdata),
	n_fish = as.integer(max(growthdata$ID2)),
	n_pheno = as.integer(max(growthdata$phenotype_num)),
	fish_id = as.integer(growthdata$ID2))
fishtable = unique(growthdata[, c("ID2", "phenotype_num")])
fishtable = fishtable[order(fishtable$ID2), ]
standata_growth$pheno_id_fish = fishtable$phenotype_num

stancode_i = ("Data/pike/growth/gm_gamma_inv_phenotype.stan")
mod_i = stan_model(stancode_i, model_name = "Individual pike growth")

fit_gamma_i = sampling(mod_i, iter = 10000, data = standata_growth, init = init_fun,
					 control = list(adapt_delta = 0.99, max_treedepth = 20), chains = 4, thin = 5)

saveRDS(fit_gamma_i, "Data/pike/growth/fit_gamma_i_phenotype.rds")

p = c("LInf", "tZero", "k", "LInf_pheno", "tZero_pheno", "k_pheno")
samps_a = as.array(fit_gamma_i, pars = p)

png("Figures/supplement/trace.png", w = 1500, h = 1500)
	bayesplot::mcmc_trace(samps_a)
dev.off()
png("Figures/supplement/intervals.png", w = 1500, h = 1500)
	bayesplot::mcmc_intervals(samps_a)
dev.off()


samps = as.matrix(fit_gamma_i, pars = p)
dat = data.frame(
	estimate = apply(samps, 2, median),
	lower = apply(samps, 2, quantile, 0.05),
	upper = apply(samps, 2, quantile, 0.95)
)
tab = cbind(rownames(dat), as.character(signif(dat$estimate, 3)), 
			paste0("(", signif(dat$lower, 3), ", ", signif(dat$upper, 3), ")"))
colnames(tab) = c("Parameter", "Posterior median", "90% CI")
knitr::kable(tab)
```

```{r run genotype model and save output, include=FALSE, eval=FALSE}
rm(list = ls())

source("Data/pike/growth/funcs.r")
growthdata <- fread("Data/Pike/growth/Pikedata_Stan.csv")

#Remove NA
growthdata <- growthdata%>%
  drop_na(genotype_num)%>%
  mutate(ID2 = as.numeric(factor(ID)))

standata_growth = list(
	Radmm = growthdata$rad_mm, 
	Agei = growthdata$lifeyear,
	N = nrow(growthdata),
	n_fish = as.integer(length(unique(growthdata$ID2))),
	n_geno = as.integer(max(growthdata$genotype_num, na.rm = T)),
	fish_id = as.integer(growthdata$ID2))
fishtable = unique(growthdata[, c("ID2", "genotype_num")])
fishtable = fishtable[order(fishtable$ID2), ]
standata_growth$geno_id_fish = fishtable$genotype_num

stancode_i = ("Data/pike/growth/gm_gamma_inv_genotype.stan")
mod_i = stan_model(stancode_i, model_name = "Individual pike growth")

fit_gamma_i = sampling(mod_i, iter = 10000, data = standata_growth, init = init_fun,
					 control = list(adapt_delta = 0.99, max_treedepth = 20), chains = 4, thin = 5)

saveRDS(fit_gamma_i, "Data/pike/growth/fit_gamma_i_genotype.rds")

p = c("LInf", "tZero", "k", "LInf_geno", "tZero_geno", "k_geno")
samps_a = as.array(fit_gamma_i, pars = p)

png("Figures/supplement/trace_geno.png", w = 1500, h = 1500)
	bayesplot::mcmc_trace(samps_a)
dev.off()
png("Figures/supplement/intervals_geno.png", w = 1500, h = 1500)
	bayesplot::mcmc_intervals(samps_a)
dev.off()


samps = as.matrix(fit_gamma_i, pars = p)
dat = data.frame(
	estimate = apply(samps, 2, median),
	lower = apply(samps, 2, quantile, 0.05),
	upper = apply(samps, 2, quantile, 0.95)
)
tab = cbind(rownames(dat), as.character(signif(dat$estimate, 3)), 
			paste0("(", signif(dat$lower, 3), ", ", signif(dat$upper, 3), ")"))
colnames(tab) = c("Parameter", "Posterior median", "90% CI")
knitr::kable(tab)
```

```{r figure VB curves phenotype, include=FALSE, eval=FALSE}
rm(list = ls())

# Ectype codes: 1 = "Brackish resident"; 2 = "Freshwater resident"; 3 = "Anadromous"; 4 = "Cross-habitat"
fit_gamma_i <- readRDS("Data/pike/growth/fit_gamma_i_phenotype.rds")
growthdata <- fread("Data/pike/growth/Pikedata_Stan.csv")

mypallette <- c("#C5D86D","#00BFB2","#F1C40F","#004643")
mylabel <- c("freshwater resident","anadromous","cross-habitat", "brackish resident")

Incpredict = function(pars, Agei) pars[1] * (1 - exp(-pars[2] * (Agei - (-pars[3]))))

pl_theme =	theme_classic() +
  theme(panel.grid.major = element_line(),
        legend.box.background = element_rect(),
        axis.line = element_line(size = .5, colour = "black", linetype = 1),
        axis.ticks = element_line(size = .5, colour = "black"),
        axis.text.x = element_text(size = 18, colour = "black"),
        axis.text.y = element_text(size = 18, colour = "black"),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        legend.text = element_text(size = 5),
        legend.title = element_text(size = 5, face = "bold"),
        legend.key.size = unit(.5,"line"),
        legend.position = c(0.3,0.8),
        plot.title = element_text(size = 20, colour = "black"))

# x is a stanfit
phenotype_predict = function(x, quants = c(0.5, 0.05, 0.95), names = c("median", "lower", "upper")) {
	pars_pheno = c("LInf_pheno", "k_pheno", "tZero_pheno")
	samps = as.matrix(x, pars = pars_pheno)

	# list of 4 phenotypes
	# each element is a matrix, 20000 rows (samples) by 15 age increments
	pheno_pr = lapply(0:3, \(i) t(apply(samps[, c(1, 5, 9) + i], 1, Incpredict, Agei = 1:15)))

	# convert to a data table summarizing by quantiles
	pheno_q = lapply(pheno_pr, \(x) {
		res = data.frame(t(apply(x, 2, quantile, quants)))
		colnames(res) = names
		res$Agei = 1:15
		res
	})
	rbindlist(pheno_q, idcol = "pheno_id")
}

fish_predict = function(x, fish_lookup) {
	pars_fish = c("LInf_i", "k_i", "tZero_i")
	samps_f = as.matrix(x, pars = pars_fish)
	fish_pr = lapply(0:119, \(i) t(apply(samps_f[, c(1, 121, 241) + i], 1, Incpredict, Agei = 1:15)))
	fish_m = lapply(fish_pr, \(x) data.frame(fit = apply(x, 2, median), Agei = 1:15))
	fish_plot = rbindlist(fish_m, idcol = "fish_id")	
	fish_plot$pheno_id = fish_lookup$phenotype_num[match(fish_plot$fish_id, fish_lookup$ID2)]
	fish_plot
}

phenotype_curves = function(x, dat) {
	fish_lookup = unique(dat[, c("ID2", "phenotype_num")])
	fish_lookup = fish_lookup[order(fish_lookup$ID2),]
	pheno_plot = phenotype_predict(x)
	fish_plot = fish_predict(x, fish_lookup)
	
pl_curves = list()
	for(i in 1:4) {
		pl_curves[[i]] = ggplot(data = pheno_plot[pheno_id == i]) + 
			geom_line(data = fish_plot[pheno_id == i], aes(x = Agei, y = fit, group = fish_id), color = "grey", size = 0.5, alpha = 0.3) + 
			geom_ribbon(aes(x = Agei, ymin = lower, ymax = upper), fill = mypallette[i], alpha = 0.5) + 
			geom_line(aes(x = Agei, y = median), color = mypallette[i], linewidth = 2) + 
		  ggtitle(mylabel[i])+
			geom_point(data = dat[dat$phenotype_num == i, ], 
					   aes(x = lifeyear, y = rad_mm, group = ID), size = 1, alpha = .9, color="black") + 
		  ylim(0,3.8)+scale_x_continuous(breaks = seq(1,15,1), labels = c("1","","3","","5","","7","","9","","11","","13","","15"))+
			pl_theme + labs(x="Age", y="otolith radius (um)")
	}
	pl_curves$nrow = 2
	do.call(grid.arrange, pl_curves)
}

phenocurves <- phenotype_curves(fit_gamma_i, growthdata)

phenocurves2 <- ggarrange(phenocurves)

phenocurves3 <- annotate_figure(phenocurves2, 
                              bottom = text_grob("Age (years)", rot = 0, size = 20), 
                              left = text_grob(expression("otolith radius"~(mu~"m")), rot = 90, size = 20))

ggsave("Figures/MS/Individual_growth_phenotypes-revised_long.png", phenocurves3, width = 15, height = 15, units = "cm", dpi = 600, bg = "white")
```

```{r figure VB curves genotype, include=FALSE, eval=FALSE}
rm(list = ls())

# Genotype codes: 1 = "putative freshwater"; 2 = "putative brackish"; 3 = "putative anadromous"; 4 = "unassigned"
fit_gamma_i <- readRDS("Data/pike/growth/fit_gamma_i_genotype.rds")
growthdata <- fread("Data/pike/growth/Pikedata_Stan.csv")

growthdata <- growthdata%>%
  drop_na(genotype_num)%>%
  mutate(ID2 = as.numeric(factor(ID)))


mypallette <- c("#C5D86D","#004643","#00BFB2","#AF3800")
mylabel <- c("putative freshwater","putative brackish","putative anadromous", "unassigned")

Incpredict = function(pars, Agei) pars[1] * (1 - exp(-pars[2] * (Agei - (-pars[3]))))

pl_theme =	theme_classic() +
  theme(panel.grid.major = element_line(),
        legend.box.background = element_rect(),
        axis.line = element_line(size = .5, colour = "black", linetype = 1),
        axis.ticks = element_line(size = .5, colour = "black"),
        axis.text.x = element_text(size = 18, colour = "black"),
        axis.text.y = element_text(size = 18, colour = "black"),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        legend.text = element_text(size = 5),
        legend.title = element_text(size = 5, face = "bold"),
        legend.key.size = unit(.5,"line"),
        legend.position = c(0.3,0.8),
        plot.title = element_text(size = 15, colour = "black"))

# x is a stanfit
genotype_predict = function(x, quants = c(0.5, 0.05, 0.95), names = c("median", "lower", "upper")) {
	pars_geno = c("LInf_geno", "k_geno", "tZero_geno")
	samps = as.matrix(x, pars = pars_geno)

	# list of 4 genotypes
	# each element is a matrix, 20000 rows (samples) by 15 age increments
	geno_pr = lapply(0:3, \(i) t(apply(samps[, c(1, 5, 9) + i], 1, Incpredict, Agei = 1:15)))

	# convert to a data table summarizing by quantiles
	geno_q = lapply(geno_pr, \(x) {
		res = data.frame(t(apply(x, 2, quantile, quants)))
		colnames(res) = names
		res$Agei = 1:15
		res
	})
	rbindlist(geno_q, idcol = "geno_id")
}

fish_predict = function(x, fish_lookup) {
	pars_fish = c("LInf_i", "k_i", "tZero_i")
	samps_f = as.matrix(x, pars = pars_fish)
	fish_pr = lapply(0:100, \(i) t(apply(samps_f[, c(1, 102, 203) + i], 1, Incpredict, Agei = 1:15)))
	fish_m = lapply(fish_pr, \(x) data.frame(fit = apply(x, 2, median), Agei = 1:15))
	fish_plot = rbindlist(fish_m, idcol = "fish_id")	
	fish_plot$geno_id = fish_lookup$genotype_num[match(fish_plot$fish_id, fish_lookup$ID2)]
	fish_plot
}

genotype_curves = function(x, dat) {
	fish_lookup = unique(dat[, c("ID2", "genotype_num")])
	fish_lookup = fish_lookup[order(fish_lookup$ID2),]
	geno_plot = genotype_predict(x)
	fish_plot = fish_predict(x, fish_lookup)
	
pl_curves = list()
	for(i in 1:4) {
		pl_curves[[i]] = ggplot(data = geno_plot[geno_id == i]) + 
			geom_line(data = fish_plot[geno_id == i], aes(x = Agei, y = fit, group = fish_id), color = "grey", 
			          size = 0.5, alpha = 0.3) + 
			geom_ribbon(aes(x = Agei, ymin = lower, ymax = upper), fill = mypallette[i], alpha = 0.5) + 
			geom_line(aes(x = Agei, y = median), color = mypallette[i], linewidth = 2) + 
		  ggtitle(mylabel[i])+
			geom_point(data = dat[dat$genotype_num == i, ], 
					   aes(x = lifeyear, y = rad_mm, group = ID), size = 1, alpha = .9, color="black") + 
		  ylim(0,3.8)+scale_x_continuous(breaks = seq(1,15,1), labels = c("1","","3","","5","","7","","9","","11","","13","","15"))+
			pl_theme + labs(x="Age", y="otolith radius (um)")
	}
	pl_curves$nrow = 2
	do.call(grid.arrange, pl_curves)
}

genocurves <- genotype_curves(fit_gamma_i, growthdata)

genocurves2 <- ggarrange(genocurves)

genocurves3 <- annotate_figure(genocurves2, 
                              bottom = text_grob("Age (years)", rot = 0, size = 20), 
                              left = text_grob(expression("otolith radius"~(mu~"m")), rot = 90, size = 20))

ggsave("Figures/supplement/Individual_growth_genotypes_revised.png", genocurves3, 
       width = 15, height = 15, units = "cm", dpi = 600, bg = "white")
```

```{r growth curve evaluation phenotype, include=FALSE, eval=FALSE}
rm(list = ls())
#Evaluation of Stan model for individual fish
fit <- readRDS("Data/pike/growth/fit_gamma_i_phenotype.rds")
#write data matrix for plotting
samples <- as.array(fit, pars=c("LInf_pheno", "k_pheno", "tZero_pheno", "phi"))
head(samples)

#write arrays for eval
sample.LInf.pheno <- as.array(fit, pars = c("LInf_pheno"))
sample.k.pheno <- as.array(fit, pars = c("k_pheno"))
sample.tZero.pheno <- as.array(fit, pars = c("tZero_pheno"))
sample.phi.pheno <- as.array(fit, pars = "phi")

#Look at parameter space with intervals
mcmc_intervals(sample.LInf.pheno)
mcmc_intervals(sample.k.pheno)
mcmc_intervals(sample.tZero.pheno)
mcmc_intervals(sample.phi.pheno)

#Look at summary stats
summary(fit, pars = c("LInf_pheno"))
summary(fit, pars = c("k_pheno"))
summary(fit, pars = c("tZero_pheno"))
summary(fit, pars = c("phi"))

#MCMC combo plots
png(filename = "Figures/supplement/histtrace_LInf_pheno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("LInf_pheno[1]", "LInf_pheno[2]", "LInf_pheno[3]", "LInf_pheno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_k_pheno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("k_pheno[1]", "k_pheno[2]", "k_pheno[3]", "k_pheno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_tZero_pheno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("tZero_pheno[1]", "tZero_pheno[2]", "tZero_pheno[3]", "tZero_pheno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_phi_pheno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("phi"))
dev.off()

#Pairs plots
png(filename = "Figures/supplement/pairs_pheno1.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_pheno[1]", "k_pheno[1]", "tZero_pheno[1]","phi"))
dev.off()

png(filename = "Figures/supplement/pairs_pheno2.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_pheno[2]", "k_pheno[2]", "tZero_pheno[2]", "phi"))
dev.off()

png(filename = "Figures/supplement/pairs_pheno3.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_pheno[3]", "k_pheno[3]", "tZero_pheno[3]", "phi"))
dev.off()

png(filename = "Figures/supplement/pairs_pheno4.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_pheno[4]", "k_pheno[4]", "tZero_pheno[4]", "phi"))
dev.off()
```

```{r growth curve evaluation genotype, include=FALSE, eval=FALSE}
rm(list = ls())
#Evaluation of Stan model for individual fish
fit <- readRDS("Data/pike/growth/fit_gamma_i_genotype.rds")
#write data matrix for plotting
samples <- as.array(fit, pars=c("LInf_geno", "k_geno", "tZero_geno", "phi"))
head(samples)

#write arrays for eval
sample.LInf.geno <- as.array(fit, pars = c("LInf_geno"))
sample.k.geno <- as.array(fit, pars = c("k_geno"))
sample.tZero.geno <- as.array(fit, pars = c("tZero_geno"))
sample.phi.geno <- as.array(fit, pars = "phi")

#Look at parameter space with intervals
mcmc_intervals(sample.LInf.geno)
mcmc_intervals(sample.k.geno)
mcmc_intervals(sample.tZero.geno)
mcmc_intervals(sample.phi.geno)

#Look at summary stats
summary(fit, pars = c("LInf_geno"))
summary(fit, pars = c("k_geno"))
summary(fit, pars = c("tZero_geno"))
summary(fit, pars = c("phi"))

#MCMC combo plots
png(filename = "Figures/supplement/histtrace_LInf_geno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("LInf_geno[1]", "LInf_geno[2]", "LInf_geno[3]", "LInf_geno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_k_geno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("k_geno[1]", "k_geno[2]", "k_geno[3]", "k_geno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_tZero_geno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("tZero_geno[1]", "tZero_geno[2]", "tZero_geno[3]", "tZero_geno[4]"))
dev.off()

png(filename = "Figures/supplement/histtrace_phi_geno.png", width = 5000, height = 2500, res = 600)
mcmc_combo(samples, c("hist", "trace"), pars = c("phi"))
dev.off()

#Pairs plots
png(filename = "Figures/supplement/pairs_geno1.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_geno[1]", "k_geno[1]", "tZero_geno[1]","phi"))
dev.off()

png(filename = "Figures/supplement/pairs_geno2.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_geno[2]", "k_geno[2]", "tZero_geno[2]", "phi"))
dev.off()

png(filename = "Figures/supplement/pairs_geno3.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_geno[3]", "k_geno[3]", "tZero_geno[3]", "phi"))
dev.off()

png(filename = "Figures/supplement/pairs_geno4.png", width = 5000, height = 2500, res = 600)
pairs(fit, pars = c("LInf_geno[4]", "k_geno[4]", "tZero_geno[4]", "phi"))
dev.off()
```

# Phenotype - genotype matching
## Testing the association of phenotype and genotype and capture location and genotype
```{r Distribution tests of phenotype-genotype match, include=FALSE, eval=FALSE}
#Script to assess the association of genotype with behavioral phenotype and capture location
#Genotypes: 1=Freshwater; 2=brackish; 3=anadromous
rm(list = ls())

#Load phenotype-genotype data with assignment probabilities
Otogen_prob <- fread("Data/pike/genetics/Phenotype-genotype-three-clusters.csv")

#Load phenotype-gentype data with discrete genotype assignments
Otogen_discrete <- fread("Data/pike/genetics/discrete-cluster-assignment.csv")

#Test if phenotype is predicted by genotype
chisq.test(table(Otogen_discrete$phenotype, Otogen_discrete$hardclust))
#phenotype highly significant

#Test if genotype is predicted by capture location
chisq.test(table(Otogen_discrete$waterbody, Otogen_discrete$hardclust))
#Capture location highly significant, note that this may include simply the probability of being captured in freshwater as
#a freshwater genotype, so not necessarily reflecting isolation by distance, but rather type of habitat

#genetic structuring by phenotype
Otogen_prob2 <- Otogen_prob%>%
  transmute(ID=ID,
            phenotype=phenotype,
            cluster1=as.numeric(cluster1),
            cluster2=as.numeric(cluster2),
            cluster3=as.numeric(cluster3))

#genetic structuring by capture location
Otogen_prob3 <- Otogen_prob%>%
  transmute(ID=ID,
            waterbody=waterbody,
            cluster1=as.numeric(cluster1),
            cluster2=as.numeric(cluster2),
            cluster3=as.numeric(cluster3))

#Turn predictor variable into factor
Otogen_prob2$phenotype <- as.factor(Otogen_prob2$phenotype)
Otogen_prob3$waterbody <- as.factor(Otogen_prob3$waterbody)

#mvabund() will only work on data frames
Otogen_prob2 <- as.data.frame(Otogen_prob2)
Otogen_prob3 <- as.data.frame(Otogen_prob3)

Genetics_pheno <- mvabund(Otogen_prob2[3:5])
Genetics_loc <- mvabund(Otogen_prob3[3:5])

boxplot(Genetics_pheno, na.rm=T)
boxplot(Genetics_loc, na.rm=T)

Gendist_pheno <- vegdist(Genetics_pheno, method = "euclidean", na.rm = T)
Gendist_loc <- vegdist(Genetics_loc, method = "euclidean", na.rm = T)

#Check out structure with PCA
PCA_pheno <- prcomp(Genetics_pheno)
PCA_loc <- prcomp(Genetics_loc)

plot(PCA_pheno)
plot(PCA_loc)

fviz_pca_biplot(PCA_pheno, repel = TRUE)
fviz_pca_biplot(PCA_loc, repel = TRUE)

plot(PCA_pheno$x[,1], PCA_pheno$x[,2], col=factor(Otogen_prob2$phenotype))
plot(PCA_loc$x[,1], PCA_loc$x[,2], col=factor(Otogen_prob3$waterbody))

#Test assumptions for PERMANOVA
Disp_pheno <- betadisper(Gendist_pheno, Otogen_prob2$phenotype)
Disp_loc <- betadisper(Gendist_loc, Otogen_prob3$waterbody)
anova(Disp_pheno)
anova(Disp_loc)
#Distribution of variances differs between groups in both cases, move to visual examination
plot(Disp_pheno)
plot(Disp_loc)
#Large variation in anadromous and freshwater genotype as well as capture location freshwater, therefore testing with
#PERMANOVA (no assumptions about multivariate normality or variance homogeneity)

adonis2(Genetics_pheno~Otogen_prob2$phenotype, method = "euclidean", permutations = 9999, na.rm=T)
#Phenotype is highly significant as predictor of genotype assignement probability and explains 51.1% of variance in genotype
#distribution (R2 = 0.511)

adonis2(Genetics_loc~Otogen_prob3$waterbody, method = "euclidean", permutations = 9999, na.rm=T)
#Waterbody is highly significant as predictor of genotype assignement probability and explains 54.7% of variance in genotype
#distribution (R2 = 0.547)

#Pairwise testing to assess between-phenotype differences in genotype distribution
pairwise.perm.manova(Genetics_pheno, Otogen_prob2$phenotype, test = c("Pillai", "Wilks",
  "Hotelling-Lawley", "Roy", "Spherical"), nperm = 999, 
  progress = TRUE, p.method = "fdr", F = FALSE, R2 = FALSE)
#All significantly different except for anadromous and freshwater resident pike (p = 0.317)

#Pairwise testing to assess between-location differences in genotype distribution
pairwise.perm.manova(Genetics_loc, Otogen_prob3$waterbody, test = c("Pillai", "Wilks",
  "Hotelling-Lawley", "Roy", "Spherical"), nperm = 999, 
  progress = TRUE, p.method = "fdr", F = FALSE, R2 = FALSE)
#No differences between lagoons, only between tributaries and all lagoons
```
## Figures
```{r phenotype-genotype matching figure assignment probability, include=FALSE, eval=FALSE}
#Script to show match between observed behavioral phenotypes (inferred by otolith microchemistry) and assignment 
#probabilities to the 3 genotypes
rm(list = ls())

#Read in data
Genotypes <- fread("Data/pike/genetics/Genotype-phenotype-combined.csv")

Genotypes2 <- Genotypes%>%
  mutate(cluster_lvl=fct_relevel(cluster,
                                 "freshwater",
                                 "anadromous",
                                 "brackish"))

#subset according to behavioral phenotype
df_F <- subset(Genotypes2, phenotype == "FW_resident")
df_A <- subset(Genotypes2, phenotype == "Anadromous")
df_T <- subset(Genotypes2, phenotype == "Trans-habitat")
df_B <- subset(Genotypes2, phenotype == "BW-resident")

#Graphic params for subplots
pl_theme1 =	theme_classic() +
	theme(panel.grid.major = element_line(),
		  legend.box.background = element_rect(),
		  axis.line = element_line(size = .5, colour = "black", linetype = 1),
		  axis.ticks = element_line(size = .5, colour = "black"),
		  axis.text.y = element_text(size = 15, colour = "black"),
		  axis.text.x = element_text(size = 15, colour = "transparent", angle = 90),
		  legend.text = element_text(size = 15),
		  axis.title.x = element_blank(),
		  axis.title.y = element_blank(),
		  plot.title = element_text(size = 15, colour = "black"),
		  plot.margin = unit(c(.8,0,-1.4,1), "cm"),
		  legend.position = "none")

pl_theme2 =	theme_classic() +
	theme(panel.grid.major = element_line(),
		  legend.box.background = element_rect(),
		  axis.line = element_line(size = .5, colour = "black", linetype = 1),
		  axis.ticks = element_line(size = .5, colour = "black"),
		  axis.text.y = element_text(size = 15, colour = "transparent"),
		  axis.text.x = element_text(size = 15, colour = "transparent", angle = 90),
		  legend.text = element_text(size = 15),
		  axis.title.x = element_blank(),
		  axis.title.y = element_blank(),
		  plot.title = element_text(size = 15, colour = "black"),
		  plot.margin = unit(c(.8,1,-1.4,0), "cm"),
		  legend.position = "none")

pl_theme3 =	theme_classic() +
	theme(panel.grid.major = element_line(),
		  legend.box.background = element_rect(),
		  axis.line = element_line(size = .5, colour = "black", linetype = 1),
		  axis.ticks = element_line(size = .5, colour = "black"),
		  axis.text.y = element_text(size = 15, colour = "black"),
		  axis.text.x = element_text(size = 15, colour = "black", angle = 90, vjust = .4, hjust = .95),
		  legend.text = element_text(size = 15),
		  axis.title.x = element_blank(),
		  axis.title.y = element_blank(),
		  plot.title = element_text(size = 15, colour = "black"),
		  plot.margin = unit(c(-0.8,0,0,1), "cm"),
		  legend.position = "none")

pl_theme4 =	theme_classic() +
	theme(panel.grid.major = element_line(),
		  legend.box.background = element_rect(),
		  axis.line = element_line(size = .5, colour = "black", linetype = 1),
		  axis.ticks = element_line(size = .5, colour = "black"),
		  axis.text.y = element_text(size = 15, colour = "transparent"),
		  axis.text.x = element_text(size = 15, colour = "black", angle = 90, vjust = .4, hjust = .95),
		  legend.text = element_text(size = 15),
		  axis.title.x = element_blank(),
		  axis.title.y = element_blank(),
		  plot.title = element_text(size = 15, colour = "black"),
		  plot.margin = unit(c(-0.8,1,0,0), "cm"),
		  legend.position = "none")

# Create stacked bar charts of genotype assignment probabilities for each phenotype
FW <- ggplot(df_F, aes(x = cluster_lvl, y = ass_prob, fill = cluster)) +
  geom_boxplot()+
  stat_summary(fun=mean, geom="point", shape=18, size=5, color="red", fill="red")+
  labs(title = "freshwater resident", x = "genotype", y = "Assignment probability (%)")+
  ylim(0,1)+
  scale_fill_manual(values = c("#00BFB2", "#004643", "#C5D86D"))+
  theme_bw()+
  pl_theme1

Ana <- ggplot(df_A, aes(x = cluster_lvl, y = ass_prob, fill = cluster)) +
  geom_boxplot() +
  stat_summary(fun=mean, geom="point", shape=18, size=5, color="red", fill="red")+
  labs(title = "anadromous", x = "Cluster", y = "Proportion") +
  ylim(0,1)+
  scale_fill_manual(values = c("#00BFB2", "#004643", "#C5D86D"))+
  theme_bw()+
  pl_theme2

Cross <- ggplot(df_T, aes(x = cluster_lvl, y = ass_prob, fill = cluster)) +
  geom_boxplot() +
  stat_summary(fun=mean, geom="point", shape=18, size=5, color="red", fill="red")+
  labs(title = "cross-habitat", x = "genotye", y = "Assignment probability (%)") +
  ylim(0,1)+
  scale_fill_manual(values = c("#00BFB2", "#004643", "#C5D86D"))+
  theme_bw()+
  pl_theme3

BW <- ggplot(df_B, aes(x = cluster_lvl, y = ass_prob, fill = cluster)) +
  geom_boxplot() +
  stat_summary(fun=mean, geom="point", shape=18, size=5, color="red", fill="red")+
  labs(title = "brackish resident", x = "genotype", y = "Proportion") +
  ylim(0,1)+
  scale_fill_manual(values = c("#00BFB2", "#004643", "#C5D86D"))+
  theme_bw()+
  pl_theme4

allplot <- ggarrange(FW, Ana, Cross, BW)
  

allplot2 <- annotate_figure(allplot, bottom = text_grob("putative genotype", color = "black", size = 15),
                left = text_grob("Assignment probability", color = "black", size = 15, rot = 90, vjust = 2))

ggsave("Figures/MS/Ecotypes_genetic_clusters_k3.png", allplot2, 
       width = 15, height = 15, units = "cm", dpi = 600, bg = "white")
```

```{r r phenotype-genotype matching figure hard-assigned genotypes, include=FALSE, eval=FALSE}
#Script to show match between observed behavioral phenotypes and discrete genotype assignements
rm(list = ls())

#Load iscrete genotype data
Genotype_discrete <- fread("Data/pike/genetics/discrete-cluster-assignment.csv")

#Figure for hard assignments
Hardclusts <- ggplot(Genotype_discrete, aes(hardclust, fill = phenotype))+geom_bar(position = "fill", width = 0.7)+theme_classic()+
  scale_x_discrete(breaks = c("1","3","2","U"), 
                   labels=c("freshwater", "anadromous", "brackish" ,"unknown"))+
  scale_y_continuous(breaks = seq(0,1,0.2), 
                     labels = c("0,0", "0,2", "0,4", "0,6", "0,8", "1,0"))+
  scale_fill_manual(breaks = c("Trans-habitat", "FW_resident", "Anadromous", "BW-resident"), 
                    labels = c("cross-habitat", "anadromous", "freshwater resident", "brackish resident"),
                    values = c("#F1C40F", "#00BFB2", "#C5D86D", "#004643"))+
  theme(axis.line = element_line(size = .5, colour = "black", linetype = 1),
		  axis.ticks = element_line(size = .5, colour = "black"),
		  axis.text.y = element_text(size = 10, colour = "black"),
		  axis.text.x = element_text(size = 10, colour = "black"),
		  legend.text = element_text(size = 10))+
  labs(x="genotype", y="proportion", fill="phenotype")

ggsave("Figures/supplement/phenotypes-genetic-clusters-k3_hard-assigned.pdf", Hardclusts, 
       width = 15, height = 9, units = "cm", dpi = 600, bg = "white")
```

